---
title: "Niger Metaphlan3 analysis"
author: "Drew Schwartz"
date: "6/17/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir ="/Users/drewschwartz/Box Sync/2022_DrewAmy_NigerSAMAbxMicrobiome")
```

```{r}
library(ggplot2)
library(vegan)
library(plyr)
library(dplyr)
library(viridis)
library(reshape2)
library(RColorBrewer)
library(multcomp)
library(readr)
library(gtools)
library(lme4)
library(sjPlot)
library(lsmeans)
library(labdsv)
library(lmerTest)
library(anthro)
library(randomForest)
library(ggExtra)

metadata <- read.delim(sep=',',"210615_Nigermetadata.csv",header=TRUE,stringsAsFactors = F)
#Remove anything under 200K reads
metadata = metadata[which(metadata$Post.processed_readcount > 200000),]
metadata = metadata[which(!metadata$Treatment %in% c("Control")),]

#Remove 2 SAMcontrol with Z score >-3 and MUAC >11.5cm
metadata <- subset(metadata, idno!="7006" & idno!="7007")

#write.csv(DF_widemeta3, "2022MetadatawithShannRich.csv")

Specieswide<-read.delim(sep='',"metaphlan3datatables/210621_wide_metaphlanStandard_species.txt",header=TRUE,stringsAsFactors = F)
Genuswide<-read.delim(sep='',"metaphlan3datatables/210621_wide_metaphlanStandard_genus.txt",header=TRUE,stringsAsFactors = F)
Familywide<-read.delim(sep='',"metaphlan3datatables/210621_wide_metaphlanStandard_family.txt",header=TRUE,stringsAsFactors = F)
Phylumwide<-read.delim(sep='',"metaphlan3datatables/210621_wide_metaphlanStandard_phyla.txt",header=TRUE,stringsAsFactors = F)
```

```{r}
#Calculate WHZ scores
Form2<-read.delim("Basic_protocols_and_surveys/Survey_results/microbiome_form2.csv", header=TRUE, sep=",")
Form9<-read.delim("Basic_protocols_and_surveys/Survey_results/microbiome_form9.csv", header=TRUE, sep=",")
Followup<-read.delim("Basic_protocols_and_surveys/Survey_results/Followup_base_amoxi_micro.csv", header=TRUE, sep=",")
Form2Z<-with(
  Form2,
  anthro_zscores(sex=BSEX, age=BAGE,weight = BWT, lenhei = BHT, is_age_in_month = TRUE
  )
)
write.csv(Form2Z,"Basic_protocols_and_surveys/Survey_results/microbiome_form2Zscore.csv")
Form9Z<-with(
  Form9,
  anthro_zscores(sex=FSEX, age=FAGE,weight = FWT, lenhei = FHT, is_age_in_month = TRUE
  )
)
write.csv(Form9Z,"Basic_protocols_and_surveys/Survey_results/microbiome_form9Zscore.csv")
FollowupZ<-with(
  Followup,
  anthro_zscores(sex=SSEX, age=SAGE,weight = SWT, lenhei = SHT, is_age_in_month = TRUE
  )
)
write.csv(FollowupZ,"Basic_protocols_and_surveys/Survey_results/followupZscore.csv")
```


```{r}
#Richness/Diversity Table
DF_wide3<-read.delim(header=T,"metaphlan3datatables/210621_alphaDiv_metaphlanStandard_species.txt")

DF_widemeta3<-join(metadata,DF_wide3, by = "Sample")

DF_widemeta3time0 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 0),]
DF_widemeta3time4 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 4),]
DF_widemeta3time1 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 1),]
DF_widemeta3time8 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 8),]
DF_widemeta3time12 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 12),]
DF_widemeta3time104 = DF_widemeta3[which(DF_widemeta3$Week_of_follow_up == 104),]

#Linear mixed effect models on Shannon Diversity with treatment and week of followup as fixed effects and idno as random effect
fitshann<- lmer(shannon~Treatment+Week_of_follow_up+ (1|idno), data=DF_widemeta3)
anova(fitshann)
Tukshann<-lsmeans(fitshann, pairwise~Treatment, adjust="tukey")
Tukshann
# $lsmeans
# Treatment   lsmean     SE  df lower.CL upper.CL
# Amoxicillin   2.06 0.0678 150    1.927     2.20
# Healthy       1.62 0.1404 419    1.344     1.90
# Placebo       2.02 0.0664 188    1.889     2.15
# SAMcontrol    1.43 0.2966 356    0.851     2.02

# Degrees-of-freedom method: kenward-roger 
# Confidence level used: 0.95 

# $contrasts TUKEY
# contrast                 estimate     SE  df t.ratio p.value
# Amoxicillin - Healthy      0.4409 0.1610 390   2.739  0.0325
# Amoxicillin - Placebo      0.0407 0.0936 161   0.435  0.9724
# Amoxicillin - SAMcontrol   0.6264 0.3069 348   2.041  0.1750
# Healthy - Placebo         -0.4002 0.1621 404  -2.469  0.0664
# Healthy - SAMcontrol       0.1855 0.3064 329   0.605  0.9303
# Placebo - SAMcontrol       0.5857 0.3075 353   1.905  0.2278
tab_model(fitshann)
##shannon
#Predictors	Estimates	CI	p
#(Intercept)	1.96	1.82 – 2.09	<0.001
#Treatment [Healthy]	-0.44	-0.76 – -0.12	0.006
#Treatment [Placebo]	-0.04	-0.22 – 0.14	0.663
#Treatment [SAMcontrol]	-0.63	-1.23 – -0.02	0.042
#Week of follow up	0.00	0.00 – 0.01	<0.001
#Random Effects
#σ2	0.30
#τ00 idno	0.19
#ICC	0.38
#N idno	202
#Observations	428
#Marginal R2 / Conditional R2	0.033 / 0.402

shannnone<-lsmeans(fitshann, pairwise~Treatment, adjust="none")
shannnone
#$contrasts NONE
# contrast                 estimate     SE  df t.ratio p.value
# Amoxicillin - Healthy      0.4409 0.1610 390   2.739  0.0064
# Amoxicillin - Placebo      0.0407 0.0936 161   0.435  0.6643
# Amoxicillin - SAMcontrol   0.6264 0.3069 348   2.041  0.0420
# Healthy - Placebo         -0.4002 0.1621 404  -2.469  0.0140
# Healthy - SAMcontrol       0.1855 0.3064 329   0.605  0.5454
# Placebo - SAMcontrol       0.5857 0.3075 353   1.905  0.0576

#Linear mixed effect models on Shannon Diversity with Description (interaction of treatment and time) as fixed effects and idno as random effect
fitshanndescription<- lmer(shannon~Description+ (1|idno), data=DF_widemeta3)
anova(fitshanndescription)
Tukshanndescription<-lsmeans(fitshanndescription, pairwise~Description, adjust="Tukey")
Tukshanndescription
# contrast TUKEY                                     estimate    SE  df t.ratio p.value
# Healthy_NA_wk104 - SAM_Amoxicillin_wk0       -0.19615 0.156 362  -1.260  0.9929
# Healthy_NA_wk104 - SAM_Amoxicillin_wk1        0.43465 0.153 350   2.850  0.2052
# Healthy_NA_wk104 - SAM_Amoxicillin_wk104     -0.62596 0.172 401  -3.646  0.0208
# Healthy_NA_wk104 - SAM_Amoxicillin_wk12      -0.12733 0.151 346  -0.843  0.9999
# Healthy_NA_wk104 - SAM_Amoxicillin_wk4       -0.02058 0.156 361  -0.132  1.0000
# Healthy_NA_wk104 - SAM_Amoxicillin_wk8       -0.12007 0.160 378  -0.750  1.0000
# Healthy_NA_wk104 - SAM_Placebo_wk0            0.27379 0.159 371   1.724  0.9062
# Healthy_NA_wk104 - SAM_Placebo_wk1           -0.06459 0.157 370  -0.410  1.0000
# Healthy_NA_wk104 - SAM_Placebo_wk104         -0.00567 0.200 413  -0.028  1.0000
# Healthy_NA_wk104 - SAM_Placebo_wk12           0.04590 0.157 366   0.293  1.0000
# Healthy_NA_wk104 - SAM_Placebo_wk4           -0.06100 0.149 348  -0.408  1.0000
# Healthy_NA_wk104 - SAM_Placebo_wk8           -0.01482 0.164 389  -0.090  1.0000
# Healthy_NA_wk104 - SAMcontrol_NA_wk104        0.18547 0.303 299   0.612  1.0000
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk1     0.63080 0.124 267   5.104  0.0001
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk104  -0.42981 0.141 240  -3.057  0.1262
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk12    0.06882 0.126 293   0.545  1.0000
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk4     0.17557 0.129 276   1.363  0.9852
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk8     0.07608 0.132 259   0.575  1.0000
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk0         0.46994 0.156 411   3.009  0.1393
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk1         0.13156 0.155 411   0.850  0.9999
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk104       0.19048 0.198 398   0.962  0.9995
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk12        0.24205 0.154 409   1.572  0.9524
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk4         0.13515 0.147 404   0.921  0.9997
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk8         0.18133 0.161 414   1.123  0.9977
# SAM_Amoxicillin_wk0 - SAMcontrol_NA_wk104     0.38162 0.302 316   1.264  0.9926
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104  -1.06061 0.143 266  -7.403  <.0001
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk12   -0.56199 0.124 308  -4.515  0.0007
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4    -0.45523 0.128 294  -3.568  0.0279
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8    -0.55472 0.131 271  -4.247  0.0024
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk0        -0.16086 0.153 408  -1.051  0.9988
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk1        -0.49925 0.152 408  -3.294  0.0637
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk104      -0.44032 0.196 400  -2.251  0.5923
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk12       -0.38875 0.151 406  -2.579  0.3576
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk4        -0.49565 0.143 398  -3.459  0.0384
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk8        -0.44947 0.158 413  -2.838  0.2098
# SAM_Amoxicillin_wk1 - SAMcontrol_NA_wk104    -0.24918 0.300 312  -0.830  0.9999
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk12  0.49862 0.144 280   3.454  0.0403
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk4   0.60538 0.147 271   4.106  0.0041
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk8   0.50589 0.150 257   3.373  0.0521
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk0       0.89975 0.172 413   5.228  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk1       0.56136 0.171 413   3.286  0.0652
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk104     0.62029 0.211 387   2.941  0.1650
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk12      0.67186 0.170 413   3.949  0.0069
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk4       0.56496 0.164 414   3.455  0.0389
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk8       0.61114 0.177 409   3.455  0.0389
# SAM_Amoxicillin_wk104 - SAMcontrol_NA_wk104   0.81143 0.310 335   2.615  0.3352
# SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk4    0.10676 0.127 300   0.842  0.9999
# SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk8    0.00726 0.128 266   0.057  1.0000
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk0        0.40112 0.152 406   2.646  0.3147
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk1        0.06274 0.150 406   0.418  1.0000
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk104      0.12167 0.194 400   0.626  1.0000
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk12       0.17324 0.149 404   1.160  0.9968
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk4        0.06634 0.142 396   0.468  1.0000
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk8        0.11251 0.157 413   0.717  1.0000
# SAM_Amoxicillin_wk12 - SAMcontrol_NA_wk104    0.31280 0.299 311   1.045  0.9989
# SAM_Amoxicillin_wk4 - SAM_Amoxicillin_wk8    -0.09949 0.132 263  -0.753  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk0         0.29437 0.156 410   1.886  0.8331
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk1        -0.04402 0.155 410  -0.285  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk104       0.01491 0.198 398   0.075  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk12        0.06648 0.154 409   0.432  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk4        -0.04042 0.147 404  -0.276  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk8         0.00576 0.161 414   0.036  1.0000
# SAM_Amoxicillin_wk4 - SAMcontrol_NA_wk104     0.20605 0.302 316   0.683  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk0         0.39386 0.161 414   2.452  0.4442
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk1         0.05548 0.159 413   0.348  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk104       0.11440 0.202 394   0.567  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk12        0.16597 0.158 413   1.047  0.9989
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk4         0.05907 0.151 411   0.390  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk8         0.10525 0.166 414   0.635  1.0000
# SAM_Amoxicillin_wk8 - SAMcontrol_NA_wk104     0.30554 0.304 322   1.005  0.9993
# SAM_Placebo_wk0 - SAM_Placebo_wk1            -0.33838 0.136 298  -2.486  0.4216
# SAM_Placebo_wk0 - SAM_Placebo_wk104          -0.27946 0.175 259  -1.593  0.9468
# SAM_Placebo_wk0 - SAM_Placebo_wk12           -0.22789 0.142 342  -1.601  0.9450
# SAM_Placebo_wk0 - SAM_Placebo_wk4            -0.33479 0.133 329  -2.517  0.3997
# SAM_Placebo_wk0 - SAM_Placebo_wk8            -0.28861 0.148 315  -1.951  0.7967
# SAM_Placebo_wk0 - SAMcontrol_NA_wk104        -0.08832 0.303 320  -0.291  1.0000
# SAM_Placebo_wk1 - SAM_Placebo_wk104           0.05893 0.181 284   0.325  1.0000
# SAM_Placebo_wk1 - SAM_Placebo_wk12            0.11050 0.139 335   0.793  0.9999
# SAM_Placebo_wk1 - SAM_Placebo_wk4             0.00360 0.132 333   0.027  1.0000
# SAM_Placebo_wk1 - SAM_Placebo_wk8             0.04977 0.145 308   0.344  1.0000
# SAM_Placebo_wk1 - SAMcontrol_NA_wk104         0.25006 0.303 319   0.826  0.9999
# SAM_Placebo_wk104 - SAM_Placebo_wk12          0.05157 0.187 314   0.275  1.0000
# SAM_Placebo_wk104 - SAM_Placebo_wk4          -0.05533 0.180 304  -0.307  1.0000
# SAM_Placebo_wk104 - SAM_Placebo_wk8          -0.00916 0.191 299  -0.048  1.0000
# SAM_Placebo_wk104 - SAMcontrol_NA_wk104       0.19114 0.327 363   0.585  1.0000
# SAM_Placebo_wk12 - SAM_Placebo_wk4           -0.10690 0.130 328  -0.824  0.9999
# SAM_Placebo_wk12 - SAM_Placebo_wk8           -0.06073 0.142 298  -0.428  1.0000
# SAM_Placebo_wk12 - SAMcontrol_NA_wk104        0.13957 0.302 318   0.462  1.0000
# SAM_Placebo_wk4 - SAM_Placebo_wk8             0.04617 0.134 290   0.344  1.0000
# SAM_Placebo_wk4 - SAMcontrol_NA_wk104         0.24647 0.299 311   0.825  0.9999
# SAM_Placebo_wk8 - SAMcontrol_NA_wk104         0.20029 0.306 327   0.654  1.0000
tab_model(fitshanndescription)

#Predictors	Estimates	CI	p
#(Intercept)	1.98	1.76 – 2.20	<0.001
#Description [SAM
#Amoxicillin wk0]	0.20	-0.11 – 0.50	0.208
#Description [SAM
#Amoxicillin wk1]	-0.43	-0.73 – -0.14	0.005
#Description [SAM
#Amoxicillin wk104]	0.63	0.29 – 0.96	<0.001
#Description [SAM
#Amoxicillin wk12]	0.13	-0.17 – 0.42	0.400
#Description [SAM
#Amoxicillin wk4]	0.02	-0.29 – 0.33	0.895
#Description [SAM
#Amoxicillin wk8]	0.12	-0.19 – 0.43	0.454
#Description [SAM Placebo
#wk0]	-0.27	-0.59 – 0.04	0.085
#Description [SAM Placebo
#wk1]	0.06	-0.24 – 0.37	0.681
#Description [SAM Placebo
#wk104]	0.01	-0.39 – 0.40	0.977
#Description [SAM Placebo
#wk12]	-0.05	-0.35 – 0.26	0.769
#Description [SAM Placebo
#wk4]	0.06	-0.23 – 0.35	0.683
#Description [SAM Placebo
#wk8]	0.01	-0.31 – 0.34	0.928
#Description [SAMcontrol
#NA wk104]	-0.19	-0.78 – 0.41	0.541
#Random Effects
#σ2	0.25
#τ00 idno	0.22
#ICC	0.47
#N idno	202
#Observations	428
#Marginal R2 / Conditional R2	0.095 / 0.520

ShannAOVDF_widemeta3time104<-aov(shannon~Treatment, data=DF_widemeta3time104)
TukeyHSD(ShannAOVDF_widemeta3time104, "Treatment", ordered= FALSE)
#Tukey multiple comparisons of means
#    95% family-wise confidence level

#Fit: aov(formula = shannon ~ Treatment, data = DF_widemeta3time104)

#Treatment
#                             diff        lwr         upr     p adj
#Healthy-Amoxicillin    -0.5709279 -1.0522340 -0.08962183 0.0135234
#Placebo-Amoxicillin    -0.3067228 -0.9352150  0.32176932 0.5769517
#SAMcontrol-Amoxicillin -0.7563976 -1.5838460  0.07105077 0.0854255
#Placebo-Healthy         0.2642051 -0.3130536  0.84146375 0.6271256
#SAMcontrol-Healthy     -0.1854697 -0.9747073  0.60376791 0.9261982
#SAMcontrol-Placebo     -0.4496748 -1.3363781  0.43702861 0.5453785

# NOW FOR RICHNESS!

#Linear mixed effect models on Richness with treatment and week of followup as fixed effects and idno as random effect
fitrich<- lmer(richness~Treatment+Week_of_follow_up+ (1|idno), data=DF_widemeta3)
anova(fitrich)
#Analysis of Variance Table
#                  npar Sum Sq Mean Sq F value
#Treatment            3 1006.5   335.5  2.2964
#Week_of_follow_up    1 9270.5  9270.5 63.4540
Tukrich<-lsmeans(fitrich, pairwise~Treatment, adjust="tukey")
Tukrich
#contrast TUKEY                 estimate   SE  df t.ratio p.value
# contrast                 estimate   SE  df t.ratio p.value
# Amoxicillin - Healthy       12.69 3.86 368   3.286  0.0061
# Amoxicillin - Placebo        3.55 2.39 171   1.482  0.4504
# Amoxicillin - SAMcontrol    18.81 7.46 314   2.521  0.0587
# Healthy - Placebo           -9.15 3.87 384  -2.363  0.0861
# Healthy - SAMcontrol         6.12 7.49 294   0.818  0.8462
# Placebo - SAMcontrol        15.27 7.47 319   2.044  0.1740
tab_model(fitrich)
#	richness
#Predictors	Estimates	CI	p
#(Intercept)	35.01	31.57 – 38.46	<0.001
#Treatment [Healthy]	-12.69	-20.28 – -5.11	0.001
#Treatment [Placebo]	-3.55	-8.24 – 1.15	0.139
#Treatment [SAMcontrol]	-18.81	-33.48 – -4.15	0.012
#Week of follow up	0.18	0.14 – 0.23	<0.001
#Random Effects
#σ2	146.10
#τ00 idno	144.54
#ICC	0.50
#N idno	202
#Observations	428
#Marginal R2 / Conditional R2	0.097 / 0.546

#Linear mixed effect models on Richness with Description (combined treatment and week of followup) as fixed effects and idno as random effect
fitrichdescription<- lmer(richness~Description+ (1|idno), data=DF_widemeta3)
anova(fitrichdescription)
#            npar Sum Sq Mean Sq F value
#Description   13  15831  1217.8  10.081
Tukrichdescription<-lsmeans(fitrichdescription, pairwise~Description, adjust="tukey")
Tukrichdescription
 
# contrast TUKEY                                estimate   SE  df t.ratio p.value
# Healthy_NA_wk104 - SAM_Amoxicillin_wk0         3.6464 3.77 335   0.968  0.9995
# Healthy_NA_wk104 - SAM_Amoxicillin_wk1        13.5491 3.70 322   3.658  0.0204
# Healthy_NA_wk104 - SAM_Amoxicillin_wk104     -19.3613 4.09 383  -4.732  0.0003
# Healthy_NA_wk104 - SAM_Amoxicillin_wk12        0.9913 3.67 317   0.270  1.0000
# Healthy_NA_wk104 - SAM_Amoxicillin_wk4         5.5797 3.77 333   1.481  0.9702
# Healthy_NA_wk104 - SAM_Amoxicillin_wk8         4.5545 3.86 352   1.181  0.9961
# Healthy_NA_wk104 - SAM_Placebo_wk0            13.4803 3.83 349   3.521  0.0318
# Healthy_NA_wk104 - SAM_Placebo_wk1             6.0857 3.80 346   1.602  0.9447
# Healthy_NA_wk104 - SAM_Placebo_wk104           4.3985 4.68 414   0.939  0.9996
# Healthy_NA_wk104 - SAM_Placebo_wk12            9.5325 3.78 342   2.520  0.3977
# Healthy_NA_wk104 - SAM_Placebo_wk4             6.1457 3.63 321   1.692  0.9174
# Healthy_NA_wk104 - SAM_Placebo_wk8             6.5769 3.93 369   1.675  0.9236
# Healthy_NA_wk104 - SAMcontrol_NA_wk104         6.1228 7.48 269   0.819  0.9999
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk1      9.9027 2.73 255   3.624  0.0236
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk104  -23.0077 3.08 232  -7.458  <.0001
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk12    -2.6551 2.81 275  -0.944  0.9996
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk4      1.9333 2.85 262   0.677  1.0000
# SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk8      0.9081 2.92 246   0.311  1.0000
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk0          9.8339 3.69 405   2.665  0.3034
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk1          2.4392 3.66 404   0.667  1.0000
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk104        0.7521 4.57 402   0.164  1.0000
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk12         5.8861 3.64 402   1.615  0.9414
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk4          2.4993 3.49 393   0.717  1.0000
# SAM_Amoxicillin_wk0 - SAM_Placebo_wk8          2.9305 3.79 412   0.773  1.0000
# SAM_Amoxicillin_wk0 - SAMcontrol_NA_wk104      2.4764 7.41 285   0.334  1.0000
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104  -32.9104 3.17 252 -10.397  <.0001
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk12   -12.5578 2.78 290  -4.513  0.0008
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4     -7.9694 2.84 277  -2.804  0.2285
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8     -8.9946 2.89 257  -3.112  0.1089
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk0         -0.0688 3.62 400  -0.019  1.0000
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk1         -7.4635 3.59 399  -2.078  0.7164
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk104       -9.1506 4.52 404  -2.025  0.7514
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk12        -4.0166 3.58 397  -1.123  0.9977
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk4         -7.4034 3.42 385  -2.167  0.6534
# SAM_Amoxicillin_wk1 - SAM_Placebo_wk8         -6.9722 3.73 409  -1.870  0.8417
# SAM_Amoxicillin_wk1 - SAMcontrol_NA_wk104     -7.4263 7.37 282  -1.007  0.9992
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk12  20.3526 3.20 263   6.358  <.0001
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk4   24.9410 3.26 256   7.647  <.0001
# SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk8   23.9159 3.30 243   7.238  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk0       32.8416 4.02 414   8.170  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk1       25.4470 3.99 414   6.376  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk104     23.7598 4.84 390   4.907  0.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk12      28.8938 3.98 414   7.265  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk4       25.5070 3.83 413   6.655  <.0001
# SAM_Amoxicillin_wk104 - SAM_Placebo_wk8       25.9382 4.11 412   6.304  <.0001
# SAM_Amoxicillin_wk104 - SAMcontrol_NA_wk104   25.4841 7.57 303   3.364  0.0526
# SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk4     4.5884 2.83 283   1.623  0.9388
# SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk8     3.5632 2.82 254   1.263  0.9926
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk0        12.4890 3.59 398   3.474  0.0366
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk1         5.0944 3.56 397   1.430  0.9779
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk104       3.4072 4.50 404   0.758  1.0000
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk12        8.5412 3.55 394   2.408  0.4762
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk4         5.1544 3.38 381   1.523  0.9628
# SAM_Amoxicillin_wk12 - SAM_Placebo_wk8         5.5856 3.70 408   1.510  0.9654
# SAM_Amoxicillin_wk12 - SAMcontrol_NA_wk104     5.1315 7.36 281   0.697  1.0000
# SAM_Amoxicillin_wk4 - SAM_Amoxicillin_wk8     -1.0252 2.92 251  -0.351  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk0          7.9006 3.69 405   2.142  0.6717
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk1          0.5059 3.66 404   0.138  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk104       -1.1812 4.57 402  -0.258  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk12         3.9528 3.64 402   1.085  0.9984
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk4          0.5660 3.48 392   0.162  1.0000
# SAM_Amoxicillin_wk4 - SAM_Placebo_wk8          0.9972 3.79 411   0.263  1.0000
# SAM_Amoxicillin_wk4 - SAMcontrol_NA_wk104      0.5431 7.40 285   0.073  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk0          8.9258 3.78 410   2.362  0.5099
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk1          1.5311 3.75 410   0.409  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk104       -0.1560 4.64 398  -0.034  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk12         4.9780 3.73 409   1.333  0.9880
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk4          1.5911 3.58 403   0.445  1.0000
# SAM_Amoxicillin_wk8 - SAM_Placebo_wk8          2.0224 3.88 414   0.521  1.0000
# SAM_Amoxicillin_wk8 - SAMcontrol_NA_wk104      1.5683 7.45 291   0.211  1.0000
# SAM_Placebo_wk0 - SAM_Placebo_wk1             -7.3947 3.04 282  -2.436  0.4570
# SAM_Placebo_wk0 - SAM_Placebo_wk104           -9.0818 3.87 245  -2.347  0.5218
# SAM_Placebo_wk0 - SAM_Placebo_wk12            -3.9478 3.21 320  -1.230  0.9943
# SAM_Placebo_wk0 - SAM_Placebo_wk4             -7.3346 2.99 309  -2.453  0.4448
# SAM_Placebo_wk0 - SAM_Placebo_wk8             -6.9034 3.31 293  -2.084  0.7116
# SAM_Placebo_wk0 - SAMcontrol_NA_wk104         -7.3575 7.44 290  -0.989  0.9994
# SAM_Placebo_wk1 - SAM_Placebo_wk104           -1.6871 4.03 265  -0.419  1.0000
# SAM_Placebo_wk1 - SAM_Placebo_wk12             3.4469 3.14 315   1.098  0.9981
# SAM_Placebo_wk1 - SAM_Placebo_wk4              0.0600 2.97 313   0.020  1.0000
# SAM_Placebo_wk1 - SAM_Placebo_wk8              0.4913 3.24 289   0.152  1.0000
# SAM_Placebo_wk1 - SAMcontrol_NA_wk104          0.0372 7.42 289   0.005  1.0000
# SAM_Placebo_wk104 - SAM_Placebo_wk12           5.1340 4.19 291   1.226  0.9944
# SAM_Placebo_wk104 - SAM_Placebo_wk4            1.7471 4.01 282   0.435  1.0000
# SAM_Placebo_wk104 - SAM_Placebo_wk8            2.1784 4.26 276   0.511  1.0000
# SAM_Placebo_wk104 - SAMcontrol_NA_wk104        1.7243 7.91 334   0.218  1.0000
# SAM_Placebo_wk12 - SAM_Placebo_wk4            -3.3868 2.92 309  -1.161  0.9967
# SAM_Placebo_wk12 - SAM_Placebo_wk8            -2.9556 3.16 281  -0.935  0.9997
# SAM_Placebo_wk12 - SAMcontrol_NA_wk104        -3.4097 7.41 288  -0.460  1.0000
# SAM_Placebo_wk4 - SAM_Placebo_wk8              0.4313 2.98 274   0.144  1.0000
# SAM_Placebo_wk4 - SAMcontrol_NA_wk104         -0.0229 7.34 281  -0.003  1.0000
# SAM_Placebo_wk8 - SAMcontrol_NA_wk104         -0.4541 7.49 297  -0.061  1.0000

```

```{r}
#Richness over time metaphlan3
DF_widemeta3$Treatment=factor(DF_widemeta3$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
ggplot(DF_widemeta3, aes(x=as.factor(Week_of_follow_up), y=richness, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Richness")+xlab("Timepoint")+ylab("Number of Species") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220705_Metaphlan3Richnessovertime.pdf")

summary$richmedian<-aggregate(richness ~ Description,DF_widemeta3, FUN=median)
summary$richIQR<-aggregate(richness ~ Description,DF_widemeta3, FUN=IQR)
summary$shannmedian<-aggregate(shannon ~ Description,DF_widemeta3, FUN=median)
summary$shannIQR<-aggregate(shannon ~ Description,DF_widemeta3, FUN=IQR)
write.csv(summary, "metaphlan3summarystatistics.csv")

#Plotting Shannon Diversity over time metaphlan3
ggplot(DF_widemeta3, aes(x=as.factor(Week_of_follow_up), y=shannon, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Shannon Diversity")+xlab("Timepoint")+ylab("Shannon Diversity") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220705_Metaphlan3ShannonDiversityovertime.pdf")
```

```{r}
# Averaged stack bar plots
Longgenus<-read.delim(sep='',header=T,"metaphlan3datatables/210621_metaphlanStandard_genus.txt",stringsAsFactors = F)
Longmetagenus<-join(Longgenus,metadata, by ="Sample")
# Remove readcounts less than 200K
Longmetagenus = Longmetagenus[which(Longmetagenus$Post.processed_readcount > 200000),]
Longmetagenus = Longmetagenus[which(!Longmetagenus$Treatment %in% c("Control")),]
# Remove 2 SAMcontrol kids without SAM
Longmetagenus <-subset(Longmetagenus, idno!="7006" & idno!="7007")

#Subset by timepoint and treatment
longmetagenustime0amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 0 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime0placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 0 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime1amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 1 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime1placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 1 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime4amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 4 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime4placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 4 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime8amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 8 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime8placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 8 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime12amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 12 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime12placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 12 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime104amox = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 104 & Longmetagenus$Treatment == "Amoxicillin"),]
longmetagenustime104placebo = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 104 & Longmetagenus$Treatment == "Placebo"),]
longmetagenustime104SAM = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 104 & Longmetagenus$Treatment == "SAMcontrol"),]
longmetagenustime104healthy = Longmetagenus[which(Longmetagenus$Week_of_follow_up == 104 & Longmetagenus$Treatment == "Healthy"),]

#Average by treatment condition within timepoint and pool <1% RA to "Other"
longmetagenustime0amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime0amox, FUN=sum)
longmetagenustime0amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime0amox, FUN=mean)
longmetagenustime0amoxave<- longmetagenustime0amoxave[which(longmetagenustime0amoxave$abundance > 0),]
longmetagenustime0amoxave$uniqID<-paste(longmetagenustime0amoxave$Week_of_follow_up,longmetagenustime0amoxave$Treatment,sep="_")
longmetagenustime0amoxave$normalabundance<-longmetagenustime0amoxave$abundance/sum(longmetagenustime0amoxave$abundance)*100

longmetagenustime0amoxaveOTHER<-ldply(
lapply(unique(longmetagenustime0amoxave$uniqID),function(x){rbind(longmetagenustime0amoxave[longmetagenustime0amoxave$normalabundance>=1 & longmetagenustime0amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime0amoxave$normalabundance[longmetagenustime0amoxave$uniqID==x & longmetagenustime0amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetagenustime0placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime0placebo, FUN=sum)
longmetagenustime0placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime0placebo, FUN=mean)
longmetagenustime0placeboave<- longmetagenustime0placeboave[which(longmetagenustime0placeboave$abundance > 0),]
longmetagenustime0placeboave$uniqID<-paste(longmetagenustime0placeboave$Week_of_follow_up,longmetagenustime0placeboave$Treatment,sep="_")
longmetagenustime0placeboave$normalabundance<-longmetagenustime0placeboave$abundance/sum(longmetagenustime0placeboave$abundance)*100

longmetagenustime0placeboaveOTHER<-ldply(
lapply(unique(longmetagenustime0placeboave$uniqID),function(x){rbind(longmetagenustime0placeboave[longmetagenustime0placeboave$normalabundance>=1 & longmetagenustime0placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime0placeboave$normalabundance[longmetagenustime0placeboave$uniqID==x & longmetagenustime0placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetagenustime1amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime1amox, FUN=sum)
longmetagenustime1amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime1amox, FUN=mean)
longmetagenustime1amoxave<- longmetagenustime1amoxave[which(longmetagenustime1amoxave$abundance > 0),]
longmetagenustime1amoxave$uniqID<-paste(longmetagenustime1amoxave$Week_of_follow_up,longmetagenustime1amoxave$Treatment,sep="_")
longmetagenustime1amoxave$normalabundance<-longmetagenustime1amoxave$abundance/sum(longmetagenustime1amoxave$abundance)*100

longmetagenustime1amoxaveOTHER<-ldply(
lapply(unique(longmetagenustime1amoxave$uniqID),function(x){rbind(longmetagenustime1amoxave[longmetagenustime1amoxave$normalabundance>=1 & longmetagenustime1amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime1amoxave$normalabundance[longmetagenustime1amoxave$uniqID==x & longmetagenustime1amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetagenustime1placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime1placebo, FUN=sum)
longmetagenustime1placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime1placebo, FUN=mean)
longmetagenustime1placeboave<- longmetagenustime1placeboave[which(longmetagenustime1placeboave$abundance > 0),]
longmetagenustime1placeboave$uniqID<-paste(longmetagenustime1placeboave$Week_of_follow_up,longmetagenustime1placeboave$Treatment,sep="_")
longmetagenustime1placeboave$normalabundance<-longmetagenustime1placeboave$abundance/sum(longmetagenustime1placeboave$abundance)*100

longmetagenustime1placeboaveOTHER<-ldply(
lapply(unique(longmetagenustime1placeboave$uniqID),function(x){rbind(longmetagenustime1placeboave[longmetagenustime1placeboave$normalabundance>=1 & longmetagenustime1placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime1placeboave$normalabundance[longmetagenustime1placeboave$uniqID==x & longmetagenustime1placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

#Now to subtract week 1 from week 0
Week1to0amoxave<-merge(longmetagenustime1amoxave, longmetagenustime0amoxave, by="taxa")
Week1to0amoxave$subtraction<-Week1to0amoxave$normalabundance.x - Week1to0amoxave$normalabundance.y
Week1to0amoxave$taxa=factor(Week1to0amoxave$taxa, levels=Week1to0amoxave$taxa[order(Week1to0amoxave$subtraction)])
#Fold change
Week1to0amoxave$FC<-Week1to0amoxave$normalabundance.x / Week1to0amoxave$normalabundance.y
Week1to0amoxave$taxa=factor(Week1to0amoxave$taxa, levels=Week1to0amoxave$taxa[order(Week1to0amoxave$FC)])
ggplot(Week1to0amoxave, aes(subtraction, taxa))+geom_point()
ggsave(file="2021Figs/220615_Week1minusweek0amox.pdf")
ggplot(Week1to0amoxave, aes(FC, taxa))+geom_point()
ggsave(file="2021Figs/220615_Week1FCweek0amox.pdf")

Week1to0placeboave<-merge(longmetagenustime1placeboave, longmetagenustime0placeboave, by="taxa")
Week1to0placeboave$subtraction<-Week1to0placeboave$normalabundance.x - Week1to0placeboave$normalabundance.y
Week1to0placeboave$taxa=factor(Week1to0placeboave$taxa, levels=Week1to0placeboave$taxa[order(Week1to0placeboave$subtraction)])
#Fold change
Week1to0placeboave$FC<-Week1to0placeboave$normalabundance.x / Week1to0placeboave$normalabundance.y
Week1to0placeboave$taxa=factor(Week1to0placeboave$taxa, levels=Week1to0placeboave$taxa[order(Week1to0placeboave$FC)])
ggplot(Week1to0placeboave, aes(subtraction, taxa))+geom_point()
ggsave(file="2021Figs/220615_Week1minusweek0placebo.pdf")
ggplot(Week1to0placeboave, aes(FC, taxa))+geom_point()
ggsave(file="2021Figs/220615_Week1FCweek0placebo.pdf")

#Now to ask what the relative abundance relative changes of amoxicillin over placebo
# Will first use the subtraction of week 1 to week 0 and then divide that by each other
week1to0amoxplacebo<-merge(Week1to0placeboave,Week1to0amoxave, by="taxa")
week1to0amoxplacebo$AmoxplaceboFC<-week1to0amoxplacebo$subtraction.y / week1to0amoxplacebo$subtraction.x
week1to0amoxplacebo$taxa=factor(week1to0amoxplacebo$taxa, levels=week1to0amoxplacebo$taxa[order(week1to0amoxplacebo$AmoxplaceboFC)])
#The above seems to distort the data. What about asking the fold change of amox to placebo at each timepoint?
week0amoxFCplacebo<-merge(longmetagenustime0amoxave, longmetagenustime0placeboave, by="taxa")






```


```{r}
# FAMILY LEVEL
# Averaged stack bar plots
Longfamily<-read.delim(sep='',header=T,"metaphlan3datatables/210621_metaphlanStandard_family.txt",stringsAsFactors = F)
Longmetafamily<-join(Longfamily,metadata, by ="Sample")
# Remove readcounts less than 200K
Longmetafamily = Longmetafamily[which(Longmetafamily$Post.processed_readcount > 200000),]
Longmetafamily = Longmetafamily[which(!Longmetafamily$Treatment %in% c("Control")),]

# Remove 2 SAMcontrol kids without SAM
Longmetafamily <-subset(Longmetafamily, idno!="7006" & idno!="7007")

#Subset by timepoint and treatment
longmetafamilytime0amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 0 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime0placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 0 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime1amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 1 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime1placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 1 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime4amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 4 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime4placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 4 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime8amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 8 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime8placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 8 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime12amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 12 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime12placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 12 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime104amox = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 104 & Longmetafamily$Treatment == "Amoxicillin"),]
longmetafamilytime104placebo = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 104 & Longmetafamily$Treatment == "Placebo"),]
longmetafamilytime104SAM = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 104 & Longmetafamily$Treatment == "SAMcontrol"),]
longmetafamilytime104healthy = Longmetafamily[which(Longmetafamily$Week_of_follow_up == 104 & Longmetafamily$Treatment == "Healthy"),]

#Average by treatment condition within timepoint and pool <1% RA to "Other"
longmetafamilytime0amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime0amox, FUN=sum)
longmetafamilytime0amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime0amox, FUN=mean)
longmetafamilytime0amoxave<- longmetafamilytime0amoxave[which(longmetafamilytime0amoxave$abundance > 0),]
longmetafamilytime0amoxave$uniqID<-paste(longmetafamilytime0amoxave$Week_of_follow_up,longmetafamilytime0amoxave$Treatment,sep="_")
longmetafamilytime0amoxave$normalabundance<-longmetafamilytime0amoxave$abundance/sum(longmetafamilytime0amoxave$abundance)*100

longmetafamilytime0amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime0amoxave$uniqID),function(x){rbind(longmetafamilytime0amoxave[longmetafamilytime0amoxave$normalabundance>=1 & longmetafamilytime0amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime0amoxave$normalabundance[longmetafamilytime0amoxave$uniqID==x & longmetafamilytime0amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)


ggplot(longmetafamilytime0amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime0placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime0placebo, FUN=sum)
longmetafamilytime0placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime0placebo, FUN=mean)
longmetafamilytime0placeboave<- longmetafamilytime0placeboave[which(longmetafamilytime0placeboave$abundance > 0),]
longmetafamilytime0placeboave$normalabundance<-longmetafamilytime0placeboave$abundance/sum(longmetafamilytime0placeboave$abundance)*100
longmetafamilytime0placeboave$uniqID<-paste(longmetafamilytime0placeboave$Week_of_follow_up,longmetafamilytime0placeboave$Treatment,sep="_")

longmetafamilytime0placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime0placeboave$uniqID),function(x){rbind(longmetafamilytime0placeboave[longmetafamilytime0placeboave$normalabundance>=1 & longmetafamilytime0placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime0placeboave$normalabundance[longmetafamilytime0placeboave$uniqID==x & longmetafamilytime0placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

#Average by treatment condition within timepoint and pool <1% RA to "Other"
longmetafamilytime1amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime1amox, FUN=sum)
longmetafamilytime1amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime1amox, FUN=mean)
longmetafamilytime1amoxave<- longmetafamilytime1amoxave[which(longmetafamilytime1amoxave$abundance > 0),]
longmetafamilytime1amoxave$uniqID<-paste(longmetafamilytime1amoxave$Week_of_follow_up,longmetafamilytime1amoxave$Treatment,sep="_")
longmetafamilytime1amoxave$normalabundance<-longmetafamilytime1amoxave$abundance/sum(longmetafamilytime1amoxave$abundance)*100

longmetafamilytime1amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime1amoxave$uniqID),function(x){rbind(longmetafamilytime1amoxave[longmetafamilytime1amoxave$normalabundance>=1 & longmetafamilytime1amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime1amoxave$normalabundance[longmetafamilytime1amoxave$uniqID==x & longmetafamilytime1amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)


ggplot(longmetafamilytime1amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime1placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime1placebo, FUN=sum)
longmetafamilytime1placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime1placebo, FUN=mean)
longmetafamilytime1placeboave<- longmetafamilytime1placeboave[which(longmetafamilytime1placeboave$abundance > 0),]
longmetafamilytime1placeboave$normalabundance<-longmetafamilytime1placeboave$abundance/sum(longmetafamilytime1placeboave$abundance)*100
longmetafamilytime1placeboave$uniqID<-paste(longmetafamilytime1placeboave$Week_of_follow_up,longmetafamilytime1placeboave$Treatment,sep="_")

longmetafamilytime1placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime1placeboave$uniqID),function(x){rbind(longmetafamilytime1placeboave[longmetafamilytime1placeboave$normalabundance>=1 & longmetafamilytime1placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime1placeboave$normalabundance[longmetafamilytime1placeboave$uniqID==x & longmetafamilytime1placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetafamilytime4amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime4amox, FUN=sum)
longmetafamilytime4amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime4amox, FUN=mean)
longmetafamilytime4amoxave<- longmetafamilytime4amoxave[which(longmetafamilytime4amoxave$abundance > 0),]
longmetafamilytime4amoxave$uniqID<-paste(longmetafamilytime4amoxave$Week_of_follow_up,longmetafamilytime4amoxave$Treatment,sep="_")
longmetafamilytime4amoxave$normalabundance<-longmetafamilytime4amoxave$abundance/sum(longmetafamilytime4amoxave$abundance)*100

longmetafamilytime4amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime4amoxave$uniqID),function(x){rbind(longmetafamilytime4amoxave[longmetafamilytime4amoxave$normalabundance>=1 & longmetafamilytime4amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime4amoxave$normalabundance[longmetafamilytime4amoxave$uniqID==x & longmetafamilytime4amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetafamilytime4amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime4placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime4placebo, FUN=sum)
longmetafamilytime4placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime4placebo, FUN=mean)
longmetafamilytime4placeboave<- longmetafamilytime4placeboave[which(longmetafamilytime4placeboave$abundance > 0),]
longmetafamilytime4placeboave$uniqID<-paste(longmetafamilytime4placeboave$Week_of_follow_up,longmetafamilytime4placeboave$Treatment,sep="_")
longmetafamilytime4placeboave$normalabundance<-longmetafamilytime4placeboave$abundance/sum(longmetafamilytime4placeboave$abundance)*100

longmetafamilytime4placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime4placeboave$uniqID),function(x){rbind(longmetafamilytime4placeboave[longmetafamilytime4placeboave$normalabundance>=1 & longmetafamilytime4placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime4placeboave$normalabundance[longmetafamilytime4placeboave$uniqID==x & longmetafamilytime4placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetafamilytime8amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime8amox, FUN=sum)
longmetafamilytime8amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime8amox, FUN=mean)
longmetafamilytime8amoxave<- longmetafamilytime8amoxave[which(longmetafamilytime8amoxave$abundance > 0),]
longmetafamilytime8amoxave$uniqID<-paste(longmetafamilytime8amoxave$Week_of_follow_up,longmetafamilytime8amoxave$Treatment,sep="_")
longmetafamilytime8amoxave$normalabundance<-longmetafamilytime8amoxave$abundance/sum(longmetafamilytime8amoxave$abundance)*100

longmetafamilytime8amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime8amoxave$uniqID),function(x){rbind(longmetafamilytime8amoxave[longmetafamilytime8amoxave$normalabundance>=1 & longmetafamilytime8amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime8amoxave$normalabundance[longmetafamilytime8amoxave$uniqID==x & longmetafamilytime8amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetafamilytime8amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime8placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime8placebo, FUN=sum)
longmetafamilytime8placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime8placebo, FUN=mean)
longmetafamilytime8placeboave<- longmetafamilytime8placeboave[which(longmetafamilytime8placeboave$abundance > 0),]
longmetafamilytime8placeboave$uniqID<-paste(longmetafamilytime8placeboave$Week_of_follow_up,longmetafamilytime8placeboave$Treatment,sep="_")
longmetafamilytime8placeboave$normalabundance<-longmetafamilytime8placeboave$abundance/sum(longmetafamilytime8placeboave$abundance)*100

longmetafamilytime8placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime8placeboave$uniqID),function(x){rbind(longmetafamilytime8placeboave[longmetafamilytime8placeboave$normalabundance>=1 & longmetafamilytime8placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime8placeboave$normalabundance[longmetafamilytime8placeboave$uniqID==x & longmetafamilytime8placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetafamilytime12amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime12amox, FUN=sum)
longmetafamilytime12amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime12amox, FUN=mean)
longmetafamilytime12amoxave<- longmetafamilytime12amoxave[which(longmetafamilytime12amoxave$abundance > 0),]
longmetafamilytime12amoxave$uniqID<-paste(longmetafamilytime12amoxave$Week_of_follow_up,longmetafamilytime12amoxave$Treatment,sep="_")
longmetafamilytime12amoxave$normalabundance<-longmetafamilytime12amoxave$abundance/sum(longmetafamilytime12amoxave$abundance)*100

longmetafamilytime12amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime12amoxave$uniqID),function(x){rbind(longmetafamilytime12amoxave[longmetafamilytime12amoxave$normalabundance>=1 & longmetafamilytime12amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime12amoxave$normalabundance[longmetafamilytime12amoxave$uniqID==x & longmetafamilytime12amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetafamilytime12amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime12placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime12placebo, FUN=sum)
longmetafamilytime12placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime12placebo, FUN=mean)
longmetafamilytime12placeboave<- longmetafamilytime12placeboave[which(longmetafamilytime12placeboave$abundance > 0),]
longmetafamilytime12placeboave$uniqID<-paste(longmetafamilytime12placeboave$Week_of_follow_up,longmetafamilytime12placeboave$Treatment,sep="_")
longmetafamilytime12placeboave$normalabundance<-longmetafamilytime12placeboave$abundance/sum(longmetafamilytime12placeboave$abundance)*100

longmetafamilytime12placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime12placeboave$uniqID),function(x){rbind(longmetafamilytime12placeboave[longmetafamilytime12placeboave$normalabundance>=1 & longmetafamilytime12placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime12placeboave$normalabundance[longmetafamilytime12placeboave$uniqID==x & longmetafamilytime12placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

```

```{r}
# Averaged stack bar plots
Longphyla<-read.delim(sep='',header=T,"metaphlan3datatables/210621_metaphlanStandard_phyla.txt",stringsAsFactors = F)
Longmetaphyla<-join(Longphyla,metadata, by ="Sample")
# Remove readcounts less than 200K
Longmetaphyla = Longmetaphyla[which(Longmetaphyla$Post.processed_readcount > 200000),]
Longmetaphyla = Longmetaphyla[which(!Longmetaphyla$Treatment %in% c("Control")),]
# Remove 2 SAMcontrol kids without SAM
Longmetaphyla <-subset(Longmetaphyla, idno!="7006" & idno!="7007")

#Subset by timepoint and treatment
longmetaphylatime0amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 0 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime0placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 0 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime1amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 1 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime1placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 1 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime4amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 4 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime4placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 4 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime8amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 8 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime8placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 8 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime12amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 12 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime12placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 12 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime104amox = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 104 & Longmetaphyla$Treatment == "Amoxicillin"),]
longmetaphylatime104placebo = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 104 & Longmetaphyla$Treatment == "Placebo"),]
longmetaphylatime104SAM = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 104 & Longmetaphyla$Treatment == "SAMcontrol"),]
longmetaphylatime104healthy = Longmetaphyla[which(Longmetaphyla$Week_of_follow_up == 104 & Longmetaphyla$Treatment == "Healthy"),]

#Average by treatment condition within timepoint and pool <1% RA to "Other"
longmetaphylatime0amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime0amox, FUN=sum)
longmetaphylatime0amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime0amox, FUN=mean)
longmetaphylatime0amoxave<- longmetaphylatime0amoxave[which(longmetaphylatime0amoxave$abundance > 0),]
longmetaphylatime0amoxave$uniqID<-paste(longmetaphylatime0amoxave$Week_of_follow_up,longmetaphylatime0amoxave$Treatment,sep="_")
longmetaphylatime0amoxave$normalabundance<-longmetaphylatime0amoxave$abundance/sum(longmetaphylatime0amoxave$abundance)*100

longmetaphylatime0amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime0amoxave$uniqID),function(x){rbind(longmetaphylatime0amoxave[longmetaphylatime0amoxave$normalabundance>=1 & longmetaphylatime0amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime0amoxave$normalabundance[longmetaphylatime0amoxave$uniqID==x & longmetaphylatime0amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime0amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime0placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime0placebo, FUN=sum)
longmetaphylatime0placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime0placebo, FUN=mean)
longmetaphylatime0placeboave<- longmetaphylatime0placeboave[which(longmetaphylatime0placeboave$abundance > 0),]
longmetaphylatime0placeboave$uniqID<-paste(longmetaphylatime0placeboave$Week_of_follow_up,longmetaphylatime0placeboave$Treatment,sep="_")
longmetaphylatime0placeboave$normalabundance<-longmetaphylatime0placeboave$abundance/sum(longmetaphylatime0placeboave$abundance)*100

longmetaphylatime0placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime0placeboave$uniqID),function(x){rbind(longmetaphylatime0placeboave[longmetaphylatime0placeboave$normalabundance>=1 & longmetaphylatime0placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime0placeboave$normalabundance[longmetaphylatime0placeboave$uniqID==x & longmetaphylatime0placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime0placeboaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime0<-rbind(longmetaphylatime0placeboaveOTHER, longmetaphylatime0amoxaveOTHER)
ggplot(longmetaphylatime0, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time0phylumaveabundance.pdf")
```
```{r}
longmetaphylatime1amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime1amox, FUN=sum)
longmetaphylatime1amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime1amox, FUN=mean)
longmetaphylatime1amoxave<- longmetaphylatime1amoxave[which(longmetaphylatime1amoxave$abundance > 0),]
longmetaphylatime1amoxave$uniqID<-paste(longmetaphylatime1amoxave$Week_of_follow_up,longmetaphylatime1amoxave$Treatment,sep="_")
longmetaphylatime1amoxave$normalabundance<-longmetaphylatime1amoxave$abundance/sum(longmetaphylatime1amoxave$abundance)*100

longmetaphylatime1amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime1amoxave$uniqID),function(x){rbind(longmetaphylatime1amoxave[longmetaphylatime1amoxave$normalabundance>=1 & longmetaphylatime1amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime1amoxave$normalabundance[longmetaphylatime1amoxave$uniqID==x & longmetaphylatime1amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime1amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime1placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime1placebo, FUN=sum)
longmetaphylatime1placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime1placebo, FUN=mean)
longmetaphylatime1placeboave<- longmetaphylatime1placeboave[which(longmetaphylatime1placeboave$abundance > 0),]
longmetaphylatime1placeboave$uniqID<-paste(longmetaphylatime1placeboave$Week_of_follow_up,longmetaphylatime1placeboave$Treatment,sep="_")
longmetaphylatime1placeboave$normalabundance<-longmetaphylatime1placeboave$abundance/sum(longmetaphylatime1placeboave$abundance)*100

longmetaphylatime1placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime1placeboave$uniqID),function(x){rbind(longmetaphylatime1placeboave[longmetaphylatime1placeboave$normalabundance>=1 & longmetaphylatime1placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime1placeboave$normalabundance[longmetaphylatime1placeboave$uniqID==x & longmetaphylatime1placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime1placeboaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime1<-rbind(longmetaphylatime1placeboaveOTHER, longmetaphylatime1amoxaveOTHER)
ggplot(longmetaphylatime1, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_time1phylumaveabundance.pdf")
```



```{r}
longmetaphylatime4amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime4amox, FUN=sum)
longmetaphylatime4amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime4amox, FUN=mean)
longmetaphylatime4amoxave<- longmetaphylatime4amoxave[which(longmetaphylatime4amoxave$abundance > 0),]
longmetaphylatime4amoxave$uniqID<-paste(longmetaphylatime4amoxave$Week_of_follow_up,longmetaphylatime4amoxave$Treatment,sep="_")
longmetaphylatime4amoxave$normalabundance<-longmetaphylatime4amoxave$abundance/sum(longmetaphylatime4amoxave$abundance)*100

longmetaphylatime4amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime4amoxave$uniqID),function(x){rbind(longmetaphylatime4amoxave[longmetaphylatime4amoxave$abundance>=1 & longmetaphylatime4amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime4amoxave$normalabundance[longmetaphylatime4amoxave$uniqID==x & longmetaphylatime4amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime4amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime4placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime4placebo, FUN=sum)
longmetaphylatime4placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime4placebo, FUN=mean)
longmetaphylatime4placeboave<- longmetaphylatime4placeboave[which(longmetaphylatime4placeboave$abundance > 0),]
longmetaphylatime4placeboave$uniqID<-paste(longmetaphylatime4placeboave$Week_of_follow_up,longmetaphylatime4placeboave$Treatment,sep="_")
longmetaphylatime4placeboave$normalabundance<-longmetaphylatime4placeboave$abundance/sum(longmetaphylatime4placeboave$abundance)*100

longmetaphylatime4placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime4placeboave$uniqID),function(x){rbind(longmetaphylatime4placeboave[longmetaphylatime4placeboave$normalabundance>=1 & longmetaphylatime4placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime4placeboave$normalabundance[longmetaphylatime4placeboave$uniqID==x & longmetaphylatime4placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime4placeboaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime4<-rbind(longmetaphylatime4placeboaveOTHER, longmetaphylatime4amoxaveOTHER)
ggplot(longmetaphylatime4, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time4phylumaveabundance.pdf")

```
```{r}
longmetaphylatime8amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime8amox, FUN=sum)
longmetaphylatime8amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime8amox, FUN=mean)
longmetaphylatime8amoxave<- longmetaphylatime8amoxave[which(longmetaphylatime8amoxave$abundance > 0),]
longmetaphylatime8amoxave$uniqID<-paste(longmetaphylatime8amoxave$Week_of_follow_up,longmetaphylatime8amoxave$Treatment,sep="_")
longmetaphylatime8amoxave$normalabundance<-longmetaphylatime8amoxave$abundance/sum(longmetaphylatime8amoxave$abundance)*100

longmetaphylatime8amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime8amoxave$uniqID),function(x){rbind(longmetaphylatime8amoxave[longmetaphylatime8amoxave$abundance>=1 & longmetaphylatime8amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime8amoxave$normalabundance[longmetaphylatime8amoxave$uniqID==x & longmetaphylatime8amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime8amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime8placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime8placebo, FUN=sum)
longmetaphylatime8placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime8placebo, FUN=mean)
longmetaphylatime8placeboave<- longmetaphylatime8placeboave[which(longmetaphylatime8placeboave$abundance > 0),]
longmetaphylatime8placeboave$uniqID<-paste(longmetaphylatime8placeboave$Week_of_follow_up,longmetaphylatime8placeboave$Treatment,sep="_")
longmetaphylatime8placeboave$normalabundance<-longmetaphylatime8placeboave$abundance/sum(longmetaphylatime8placeboave$abundance)*100

longmetaphylatime8placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime8placeboave$uniqID),function(x){rbind(longmetaphylatime8placeboave[longmetaphylatime8placeboave$normalabundance>=1 & longmetaphylatime8placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime8placeboave$normalabundance[longmetaphylatime8placeboave$uniqID==x & longmetaphylatime8placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime8placeboaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime8<-rbind(longmetaphylatime8placeboaveOTHER, longmetaphylatime8amoxaveOTHER)
ggplot(longmetaphylatime8, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time8phylumaveabundance.pdf")

```
```{r}
longmetaphylatime12amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime12amox, FUN=sum)
longmetaphylatime12amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime12amox, FUN=mean)
longmetaphylatime12amoxave<- longmetaphylatime12amoxave[which(longmetaphylatime12amoxave$abundance > 0),]
longmetaphylatime12amoxave$uniqID<-paste(longmetaphylatime12amoxave$Week_of_follow_up,longmetaphylatime12amoxave$Treatment,sep="_")
longmetaphylatime12amoxave$normalabundance<-longmetaphylatime12amoxave$abundance/sum(longmetaphylatime12amoxave$abundance)*100

longmetaphylatime12amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime12amoxave$uniqID),function(x){rbind(longmetaphylatime12amoxave[longmetaphylatime12amoxave$abundance>=1 & longmetaphylatime12amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime12amoxave$normalabundance[longmetaphylatime12amoxave$uniqID==x & longmetaphylatime12amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime12amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime12placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime12placebo, FUN=sum)
longmetaphylatime12placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime12placebo, FUN=mean)
longmetaphylatime12placeboave<- longmetaphylatime12placeboave[which(longmetaphylatime12placeboave$abundance > 0),]
longmetaphylatime12placeboave$uniqID<-paste(longmetaphylatime12placeboave$Week_of_follow_up,longmetaphylatime12placeboave$Treatment,sep="_")
longmetaphylatime12placeboave$normalabundance<-longmetaphylatime12placeboave$abundance/sum(longmetaphylatime12placeboave$abundance)*100

longmetaphylatime12placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime12placeboave$uniqID),function(x){rbind(longmetaphylatime12placeboave[longmetaphylatime12placeboave$normalabundance>=1 & longmetaphylatime12placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime12placeboave$normalabundance[longmetaphylatime12placeboave$uniqID==x & longmetaphylatime12placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime12placeboaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime12<-rbind(longmetaphylatime12placeboaveOTHER, longmetaphylatime12amoxaveOTHER)
ggplot(longmetaphylatime12, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time12phylumaveabundance.pdf")

```

```{r}
#Timepoint 104 at phyla level
longmetaphylatime104amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime104amox, FUN=sum)
longmetaphylatime104amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime104amox, FUN=mean)
longmetaphylatime104amoxave<- longmetaphylatime104amoxave[which(longmetaphylatime104amoxave$abundance > 0),]
longmetaphylatime104amoxave$uniqID<-paste(longmetaphylatime104amoxave$Week_of_follow_up,longmetaphylatime104amoxave$Treatment,sep="_")
longmetaphylatime104amoxave$normalabundance<-longmetaphylatime104amoxave$abundance/sum(longmetaphylatime104amoxave$abundance)*100

longmetaphylatime104amoxaveOTHER<-ldply(
lapply(unique(longmetaphylatime104amoxave$uniqID),function(x){rbind(longmetaphylatime104amoxave[longmetaphylatime104amoxave$normalabundance>=1 & longmetaphylatime104amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime104amoxave$normalabundance[longmetaphylatime104amoxave$uniqID==x & longmetaphylatime104amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetaphylatime104amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetaphylatime104placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime104placebo, FUN=sum)
longmetaphylatime104placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime104placebo, FUN=mean)
longmetaphylatime104placeboave<- longmetaphylatime104placeboave[which(longmetaphylatime104placeboave$abundance > 0),]
longmetaphylatime104placeboave$uniqID<-paste(longmetaphylatime104placeboave$Week_of_follow_up,longmetaphylatime104placeboave$Treatment,sep="_")
longmetaphylatime104placeboave$normalabundance<-longmetaphylatime104placeboave$abundance/sum(longmetaphylatime104placeboave$abundance)*100

longmetaphylatime104placeboaveOTHER<-ldply(
lapply(unique(longmetaphylatime104placeboave$uniqID),function(x){rbind(longmetaphylatime104placeboave[longmetaphylatime104placeboave$normalabundance>=1 & longmetaphylatime104placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime104placeboave$normalabundance[longmetaphylatime104placeboave$uniqID==x & longmetaphylatime104placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetaphylatime104SAMave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime104SAM, FUN=sum)
longmetaphylatime104SAMave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime104SAM, FUN=mean)
longmetaphylatime104SAMave<- longmetaphylatime104SAMave[which(longmetaphylatime104SAMave$abundance > 0),]
longmetaphylatime104SAMave$uniqID<-paste(longmetaphylatime104SAMave$Week_of_follow_up,longmetaphylatime104SAMave$Treatment,sep="_")
longmetaphylatime104SAMave$normalabundance<-longmetaphylatime104SAMave$abundance/sum(longmetaphylatime104SAMave$abundance)*100

longmetaphylatime104SAMaveOTHER<-ldply(
lapply(unique(longmetaphylatime104SAMave$uniqID),function(x){rbind(longmetaphylatime104SAMave[longmetaphylatime104SAMave$normalabundance>=1 & longmetaphylatime104SAMave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime104SAMave$normalabundance[longmetaphylatime104SAMave$uniqID==x & longmetaphylatime104SAMave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetaphylatime104healthyave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetaphylatime104healthy, FUN=sum)
longmetaphylatime104healthyave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetaphylatime104healthy, FUN=mean)
longmetaphylatime104healthyave<- longmetaphylatime104healthyave[which(longmetaphylatime104healthyave$abundance > 0),]
longmetaphylatime104healthyave$uniqID<-paste(longmetaphylatime104healthyave$Week_of_follow_up,longmetaphylatime104healthyave$Treatment,sep="_")
longmetaphylatime104healthyave$normalabundance<-longmetaphylatime104healthyave$abundance/sum(longmetaphylatime104healthyave$abundance)*100

longmetaphylatime104healthyaveOTHER<-ldply(
lapply(unique(longmetaphylatime104healthyave$uniqID),function(x){rbind(longmetaphylatime104healthyave[longmetaphylatime104healthyave$normalabundance>=1 & longmetaphylatime104healthyave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetaphylatime104healthyave$normalabundance[longmetaphylatime104healthyave$uniqID==x & longmetaphylatime104healthyave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)


longmetaphylatime104<-rbind(longmetaphylatime104placeboaveOTHER, longmetaphylatime104amoxaveOTHER, longmetaphylatime104SAMaveOTHER,  longmetaphylatime104healthyaveOTHER)
ggplot(longmetaphylatime104, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("104_Amoxicillin", "104_Placebo", "104_Healthy", "104_SAMcontrol"))
ggsave(file="2021Figs/220705_Time104phylaaveabundance.pdf")
write.csv(longmetaphylatime104, "220721_timepoint104averelabsphylum.csv")
```

```{r}
#Timepoint 104 at the Family level
longmetafamilytime104amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime104amox, FUN=sum)
longmetafamilytime104amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime104amox, FUN=mean)
longmetafamilytime104amoxave<- longmetafamilytime104amoxave[which(longmetafamilytime104amoxave$abundance > 0),]
longmetafamilytime104amoxave$uniqID<-paste(longmetafamilytime104amoxave$Week_of_follow_up,longmetafamilytime104amoxave$Treatment,sep="_")
longmetafamilytime104amoxave$normalabundance<-longmetafamilytime104amoxave$abundance/sum(longmetafamilytime104amoxave$abundance)*100

longmetafamilytime104amoxaveOTHER<-ldply(
lapply(unique(longmetafamilytime104amoxave$uniqID),function(x){rbind(longmetafamilytime104amoxave[longmetafamilytime104amoxave$normalabundance>=1 & longmetafamilytime104amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime104amoxave$normalabundance[longmetafamilytime104amoxave$uniqID==x & longmetafamilytime104amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetafamilytime104amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetafamilytime104placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime104placebo, FUN=sum)
longmetafamilytime104placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime104placebo, FUN=mean)
longmetafamilytime104placeboave<- longmetafamilytime104placeboave[which(longmetafamilytime104placeboave$abundance > 0),]
longmetafamilytime104placeboave$uniqID<-paste(longmetafamilytime104placeboave$Week_of_follow_up,longmetafamilytime104placeboave$Treatment,sep="_")
longmetafamilytime104placeboave$normalabundance<-longmetafamilytime104placeboave$abundance/sum(longmetafamilytime104placeboave$abundance)*100

longmetafamilytime104placeboaveOTHER<-ldply(
lapply(unique(longmetafamilytime104placeboave$uniqID),function(x){rbind(longmetafamilytime104placeboave[longmetafamilytime104placeboave$normalabundance>=1 & longmetafamilytime104placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime104placeboave$normalabundance[longmetafamilytime104placeboave$uniqID==x & longmetafamilytime104placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetafamilytime104SAMave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime104SAM, FUN=sum)
longmetafamilytime104SAMave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime104SAM, FUN=mean)
longmetafamilytime104SAMave<- longmetafamilytime104SAMave[which(longmetafamilytime104SAMave$abundance > 0),]
longmetafamilytime104SAMave$uniqID<-paste(longmetafamilytime104SAMave$Week_of_follow_up,longmetafamilytime104SAMave$Treatment,sep="_")
longmetafamilytime104SAMave$normalabundance<-longmetafamilytime104SAMave$abundance/sum(longmetafamilytime104SAMave$abundance)*100

longmetafamilytime104SAMaveOTHER<-ldply(
lapply(unique(longmetafamilytime104SAMave$uniqID),function(x){rbind(longmetafamilytime104SAMave[longmetafamilytime104SAMave$normalabundance>=1 & longmetafamilytime104SAMave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime104SAMave$normalabundance[longmetafamilytime104SAMave$uniqID==x & longmetafamilytime104SAMave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetafamilytime104healthyave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetafamilytime104healthy, FUN=sum)
longmetafamilytime104healthyave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetafamilytime104healthy, FUN=mean)
longmetafamilytime104healthyave<- longmetafamilytime104healthyave[which(longmetafamilytime104healthyave$abundance > 0),]
longmetafamilytime104healthyave$uniqID<-paste(longmetafamilytime104healthyave$Week_of_follow_up,longmetafamilytime104healthyave$Treatment,sep="_")
longmetafamilytime104healthyave$normalabundance<-longmetafamilytime104healthyave$abundance/sum(longmetafamilytime104healthyave$abundance)*100

longmetafamilytime104healthyaveOTHER<-ldply(
lapply(unique(longmetafamilytime104healthyave$uniqID),function(x){rbind(longmetafamilytime104healthyave[longmetafamilytime104healthyave$normalabundance>=1 & longmetafamilytime104healthyave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetafamilytime104healthyave$normalabundance[longmetafamilytime104healthyave$uniqID==x & longmetafamilytime104healthyave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)


longmetafamilytime104<-rbind(longmetafamilytime104placeboaveOTHER, longmetafamilytime104amoxaveOTHER, longmetafamilytime104SAMaveOTHER,  longmetafamilytime104healthyaveOTHER)
ggplot(longmetafamilytime104, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time104familyaveabundance.pdf")
write.csv(longmetafamilytime104, "220721_timepoint104averelabsFamily.csv")
```

```{r}
#Timepoint 104 at genus level
longmetagenustime104amoxave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime104amox, FUN=sum)
longmetagenustime104amoxave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime104amox, FUN=mean)
longmetagenustime104amoxave<- longmetagenustime104amoxave[which(longmetagenustime104amoxave$abundance > 0),]
longmetagenustime104amoxave$uniqID<-paste(longmetagenustime104amoxave$Week_of_follow_up,longmetagenustime104amoxave$Treatment,sep="_")
longmetagenustime104amoxave$normalabundance<-longmetagenustime104amoxave$abundance/sum(longmetagenustime104amoxave$abundance)*100

longmetagenustime104amoxaveOTHER<-ldply(
lapply(unique(longmetagenustime104amoxave$uniqID),function(x){rbind(longmetagenustime104amoxave[longmetagenustime104amoxave$normalabundance>=1 & longmetagenustime104amoxave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime104amoxave$normalabundance[longmetagenustime104amoxave$uniqID==x & longmetagenustime104amoxave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

ggplot(longmetagenustime104amoxaveOTHER, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))

longmetagenustime104placeboave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime104placebo, FUN=sum)
longmetagenustime104placeboave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime104placebo, FUN=mean)
longmetagenustime104placeboave<- longmetagenustime104placeboave[which(longmetagenustime104placeboave$abundance > 0),]
longmetagenustime104placeboave$uniqID<-paste(longmetagenustime104placeboave$Week_of_follow_up,longmetagenustime104placeboave$Treatment,sep="_")
longmetagenustime104placeboave$normalabundance<-longmetagenustime104placeboave$abundance/sum(longmetagenustime104placeboave$abundance)*100

longmetagenustime104placeboaveOTHER<-ldply(
lapply(unique(longmetagenustime104placeboave$uniqID),function(x){rbind(longmetagenustime104placeboave[longmetagenustime104placeboave$normalabundance>=1 & longmetagenustime104placeboave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime104placeboave$normalabundance[longmetagenustime104placeboave$uniqID==x & longmetagenustime104placeboave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetagenustime104SAMave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime104SAM, FUN=sum)
longmetagenustime104SAMave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime104SAM, FUN=mean)
longmetagenustime104SAMave<- longmetagenustime104SAMave[which(longmetagenustime104SAMave$abundance > 0),]
longmetagenustime104SAMave$uniqID<-paste(longmetagenustime104SAMave$Week_of_follow_up,longmetagenustime104SAMave$Treatment,sep="_")
longmetagenustime104SAMave$normalabundance<-longmetagenustime104SAMave$abundance/sum(longmetagenustime104SAMave$abundance)*100

longmetagenustime104SAMaveOTHER<-ldply(
lapply(unique(longmetagenustime104SAMave$uniqID),function(x){rbind(longmetagenustime104SAMave[longmetagenustime104SAMave$normalabundance>=1 & longmetagenustime104SAMave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime104SAMave$normalabundance[longmetagenustime104SAMave$uniqID==x & longmetagenustime104SAMave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)

longmetagenustime104healthyave<- aggregate(abundance~Sample + Week_of_follow_up + taxa + Treatment, data= longmetagenustime104healthy, FUN=sum)
longmetagenustime104healthyave<- aggregate(abundance~taxa + Week_of_follow_up + Treatment, data= longmetagenustime104healthy, FUN=mean)
longmetagenustime104healthyave<- longmetagenustime104healthyave[which(longmetagenustime104healthyave$abundance > 0),]
longmetagenustime104healthyave$uniqID<-paste(longmetagenustime104healthyave$Week_of_follow_up,longmetagenustime104healthyave$Treatment,sep="_")
longmetagenustime104healthyave$normalabundance<-longmetagenustime104healthyave$abundance/sum(longmetagenustime104healthyave$abundance)*100

longmetagenustime104healthyaveOTHER<-ldply(
lapply(unique(longmetagenustime104healthyave$uniqID),function(x){rbind(longmetagenustime104healthyave[longmetagenustime104healthyave$normalabundance>=1 & longmetagenustime104healthyave$uniqID==x,c("taxa","Week_of_follow_up","Treatment","normalabundance","uniqID")],
                                                data.frame(taxa=c("Other"),
                                                Week_of_follow_up=c(strsplit(x,"_")[[1]][1]),
                                                Treatment=strsplit(x,"_")[[1]][2],
                                                normalabundance=sum(longmetagenustime104healthyave$normalabundance[longmetagenustime104healthyave$uniqID==x & longmetagenustime104healthyave$normalabundance<1]),
                                                uniqID=x
                                                ))})
)


longmetagenustime104<-rbind(longmetagenustime104placeboaveOTHER, longmetagenustime104amoxaveOTHER, longmetagenustime104SAMaveOTHER,  longmetagenustime104healthyaveOTHER)
ggplot(longmetagenustime104, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
ggsave(file="2021Figs/211216_Time104genusaveabundance.pdf")
write.csv(longmetagenustime104, "220721_Timepoint104relabsgenus.csv")
```

```{r}
# Plot abundance trajectory of amoxicillin and placebo groups over time
Longmetaamoxphyla<-rbind(longmetaphylatime0amoxaveOTHER, longmetaphylatime1amoxaveOTHER, longmetaphylatime4amoxaveOTHER, longmetaphylatime8amoxaveOTHER, longmetaphylatime12amoxaveOTHER, longmetaphylatime104amoxaveOTHER)
ggplot(Longmetaamoxphyla, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("0_Amoxicillin","1_Amoxicillin", "4_Amoxicillin", "8_Amoxicillin", "12_Amoxicillin", "104_Amoxicillin"))
ggsave(file="2021Figs/211216_phylaamoxrelabsovertime.pdf")

Longmetaplacebophyla<-rbind(longmetaphylatime0placeboaveOTHER, longmetaphylatime1placeboaveOTHER, longmetaphylatime4placeboaveOTHER, longmetaphylatime8placeboaveOTHER, longmetaphylatime12placeboaveOTHER, longmetaphylatime104placeboaveOTHER)
longmetaphylaplaceboamox<-rbind(Longmetaplacebophyla, Longmetaamoxphyla)
ggplot(Longmetaplacebophyla, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("0_Placebo","1_Placebo", "4_Placebo", "8_Placebo", "12_Placebo", "104_Placebo"))
ggsave(file="2021Figs/211216_phylaplaceborelabsovertime.pdf")

ggplot(longmetaphylaplaceboamox, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ facet_wrap("Treatment", scales = "free_x")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("0_Placebo","1_Placebo", "4_Placebo", "8_Placebo", "12_Placebo", "104_Placebo", "0_Amoxicillin","1_Amoxicillin", "4_Amoxicillin", "8_Amoxicillin", "12_Amoxicillin", "104_Amoxicillin"))
ggsave(file="2021Figs/220110_phylaplaceborelabsovertime.pdf")

longmetaamoxfamily<-rbind(longmetafamilytime0amoxaveOTHER, longmetafamilytime1amoxaveOTHER, longmetafamilytime4amoxaveOTHER, longmetafamilytime8amoxaveOTHER, longmetafamilytime12amoxaveOTHER, longmetafamilytime104amoxaveOTHER)
ggplot(longmetaamoxfamily, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("0_Amoxicillin","1_Amoxicillin", "4_Amoxicillin", "8_Amoxicillin", "12_Amoxicillin", "104_Amoxicillin"))
ggsave(file="2021Figs/211216_familyamoxrelabsovertime.pdf")

longmetaplacebofamily<-rbind(longmetafamilytime0placeboaveOTHER,longmetafamilytime1placeboaveOTHER, longmetafamilytime4placeboaveOTHER, longmetafamilytime8placeboaveOTHER, longmetafamilytime12placeboaveOTHER, longmetafamilytime104placeboaveOTHER)
ggplot(longmetaplacebofamily, aes(uniqID, normalabundance, fill=taxa)) + geom_bar(stat = "identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_discrete(limits=c("0_Placebo","1_Placebo", "4_Placebo", "8_Placebo", "12_Placebo", "104_Placebo"))
ggsave(file="2021Figs/211216_familyplaceborelabsovertime.pdf")

#Write output of average abundances to file
write.csv(longmetaphylaplaceboamox, "220721_averelabsphyla.csv")
```


```{r}
#metadata analysis
metadata = metadata[which(!metadata$Treatment %in% c("Control")),]
metadata=metadata[which(metadata$Post.processed_readcount > 200000),]
metadata=subset(metadata, idno!="7006" & idno!="7007")
metadatatime0=metadata[which(metadata$Week_of_follow_up ==0),]
time0amox<-subset(metadatatime0, Treatment=='Amoxicillin')
time0placebo<-subset(metadatatime0, Treatment=='Placebo')
metadatatime1=metadata[which(metadata$Week_of_follow_up==1),]
time1amox<-subset(metadatatime1, Treatment=='Amoxicillin')
time1placebo<-subset(metadatatime1, Treatment=='Placebo')
metadatatime4=metadata[which(metadata$Week_of_follow_up==4),]
time4amox<-subset(metadatatime4, Treatment=='Amoxicillin')
time4placebo<-subset(metadatatime4, Treatment=='Placebo')
metadatatime8=metadata[which(metadata$Week_of_follow_up==8),]
time8amox<-subset(metadatatime8, Treatment=='Amoxicillin')
time8placebo<-subset(metadatatime8, Treatment=='Placebo')
metadatatime12=metadata[which(metadata$Week_of_follow_up==12),]
time12amox<-subset(metadatatime12, Treatment=='Amoxicillin')
time12placebo<-subset(metadatatime12, Treatment=='Placebo')
metadatatime104=metadata[which(metadata$Week_of_follow_up ==104),]
time104amox<-subset(metadatatime104, Treatment=='Amoxicillin')
time104placebo<-subset(metadatatime104, Treatment=='Placebo')
time104healthy<-subset(metadatatime104, Treatment=='Healthy')
time104SAM<-subset(metadatatime104, Treatment=='SAMcontrol')

pairwise.wilcox.test(metadatatime0$BAGE, metadatatime0$Treatment)
#p = 0.87
median(time0amox$BAGE)
#11
IQR(time0amox$BAGE)
#13.5
median(time0placebo$BAGE)
#11.5
IQR(time0placebo$BAGE)
#11.5
table(time0amox$BSEX)
#16M and 19F (%M = 46%)
table(time0placebo$BSEX)
#20M and 12F (%M = 62.5%)
chisq.test(metadatatime0$BSEX, metadatatime0$Treatment)
#p=0.258
pairwise.wilcox.test(metadatatime0$BMUAC, metadatatime0$Treatment)
#p=0.43
median(time0amox$BMUAC)
#11.3
IQR(time0amox$BMUAC)
#0.4
median(time0placebo$BMUAC)
#11.15
IQR(time0placebo$BMUAC)
#0.6
pairwise.wilcox.test(metadatatime0$BWFHZ, metadatatime0$Treatment)
#p=0.73
median(time0amox$BWFHZ)
#-3.17
IQR(time0amox$BWFHZ)
#0.595
median(time0placebo$BWFHZ)
#-3.115
IQR(time0placebo$BWFHZ)
#0.835

pairwise.wilcox.test(metadatatime1$visAge, metadatatime1$Treatment)
#p = 0.75
median(time1amox$visAge)
#12.25
IQR(time1amox$visAge)
#13
median(time1placebo$visAge)
#12.25
IQR(time1placebo$visAge)
#11
table(time1amox$BSEX)
#17M and 22F (%M = 44%)
table(time1placebo$BSEX)
#20M and 13F (%M = 61%)
chisq.test(metadatatime1$BSEX, metadatatime1$Treatment)
#p=0.2291
pairwise.wilcox.test(as.numeric(metadatatime1$visMUAC), metadatatime1$Treatment)
#p=0.1
median(time1amox$visMUAC)
#11.7
IQR(time1amox$visMUAC)
#0.65
median(time1placebo$visMUAC)
#11.4
IQR(time1placebo$visMUAC)
#0.8
pairwise.wilcox.test(metadatatime1$visdelMUAC, metadatatime1$Treatment)
#p=0.02
median(time1amox$visdelMUAC)
#0.4
IQR(time1amox$visdelMUAC)
#0.3
median(time1placebo$visdelMUAC)
#0.3
IQR(time1placebo$visdelMUAC)
#0.4
pairwise.wilcox.test(metadatatime1$visWHZ, metadatatime1$Treatment)
#p=0.23
median(time1amox$visWHZ)
#-2.13
IQR(time1amox$visWHZ)
#0.795
median(time1placebo$visWHZ)
#-2.19
IQR(time1placebo$visWHZ)
#1.29
pairwise.wilcox.test(metadatatime1$visdelWHZ, metadatatime1$Treatment)
#p=0.02
median(time1amox$visdelWHZ)
#1.06
IQR(time1amox$visdelWHZ)
#0.595
median(time1placebo$visdelWHZ)
#0.63
IQR(time1placebo$visdelWHZ)
#0.97



pairwise.wilcox.test(metadatatime4$visAge, metadatatime4$Treatment)
#p = 0.82
median(time4amox$visAge)
#12
IQR(time4amox$visAge)
#11
median(time4placebo$visAge)
#12
IQR(time4placebo$visAge)
#12
table(time4amox$BSEX)
#14M and 21F (%M = 44%)
table(time4placebo$BSEX)
#26M and 17F (%M = 64%)
chisq.test(metadatatime4$BSEX, metadatatime4$Treatment)
#p=0.1162
pairwise.wilcox.test(as.numeric(metadatatime4$visMUAC), metadatatime4$Treatment)
#p=0.21
median(time4amox$visMUAC)
#12
IQR(time4amox$visMUAC)
#0.55
median(time4placebo$visMUAC)
#11.6
IQR(time4placebo$visMUAC)
#1.15
pairwise.wilcox.test(metadatatime4$visWHZ, metadatatime4$Treatment)
#p=0.43
median(time4amox$visWHZ)
#-1.43
IQR(time4amox$visWHZ)
#1.155
median(time4placebo$visWHZ)
#-1.6
IQR(time4placebo$visWHZ)
#1.25
pairwise.wilcox.test(as.numeric(metadatatime4$visdelMUAC), metadatatime4$Treatment)
#p=0.44
median(time4amox$visdelMUAC)
#0.7
IQR(time4amox$visdelMUAC)
#0.5
median(time4placebo$visdelMUAC)
#0.6
IQR(time4placebo$visdelMUAC)
#0.85
pairwise.wilcox.test(as.numeric(metadatatime4$visdelWHZ), metadatatime4$Treatment)
#p=0.27
median(time4amox$visdelWHZ)
#1.46
IQR(time4amox$visdelWHZ)
#0.815
median(time4placebo$visdelWHZ)
#1.28
IQR(time4placebo$visdelWHZ)
#1.195

pairwise.wilcox.test(metadatatime8$visAge, metadatatime8$Treatment)
#p = 0.37
median(time8amox$visAge)
#14
IQR(time8amox$visAge)
#12
median(time8placebo$visAge)
#13
IQR(time8placebo$visAge)
#13
table(time8amox$BSEX)
#14M and 16F (%M = 47%)
table(time8placebo$BSEX)
#15M and 12F (%M = 56%)
chisq.test(metadatatime8$BSEX, metadatatime8$Treatment)
#p=0.6855
pairwise.wilcox.test(as.numeric(metadatatime8$visMUAC), metadatatime8$Treatment, na.rm = TRUE)
#p=6.4e-6
median(as.numeric(time8amox$visMUAC))
#12.5
IQR(time8amox$visMUAC)
#0.775
median(time8placebo$visMUAC, na.rm = TRUE)
#11.9
IQR(time8placebo$visMUAC, na.rm = TRUE)
#0.4
pairwise.wilcox.test(metadatatime8$visWHZ, metadatatime8$Treatment, na.rm = TRUE)
#0.024
median(time8amox$visWHZ)
#-1.02
IQR(time8amox$visWHZ)
#0.9075
median(time8placebo$visWHZ, na.rm = TRUE)
#-1.415
IQR(time8placebo$visWHZ, na.rm = TRUE)
#1.05
pairwise.wilcox.test(as.numeric(metadatatime8$visdelMUAC), metadatatime8$Treatment, na.rm = TRUE)
#p=0.018
median(as.numeric(time8amox$visdelMUAC))
#1.15
IQR(time8amox$visdelMUAC)
#0.6
median(time8placebo$visdelMUAC, na.rm = TRUE)
#0.9
IQR(time8placebo$visdelMUAC, na.rm = TRUE)
#0.725
pairwise.wilcox.test(metadatatime8$visdelWHZ, metadatatime8$Treatment, na.rm = TRUE)
#0.04
median(time8amox$visdelWHZ)
#2.275
IQR(time8amox$visdelWHZ)
#0.9225
median(time8placebo$visdelWHZ, na.rm = TRUE)
#1.67
IQR(time8placebo$visdelWHZ, na.rm = TRUE)
#1.63


pairwise.wilcox.test(metadatatime12$visAge, metadatatime12$Treatment)
#p = 0.04
median(time12amox$visAge)
#15
IQR(time12amox$visAge)
#13
median(time12placebo$visAge)
#13
IQR(time12placebo$visAge)
#7.5
table(time12amox$BSEX)
#19M and 22F (%M = 46%)
table(time12placebo$BSEX)
#18M and 16F (%M = 53%)
chisq.test(metadatatime12$BSEX, metadatatime12$Treatment)
#p=.736
pairwise.wilcox.test(as.numeric(metadatatime12$visMUAC), metadatatime12$Treatment)
#p=0.017
median(time12amox$visMUAC)
#12.5
IQR(time12amox$visMUAC)
#1.2
median(time12placebo$visMUAC)
#12.3
IQR(time12placebo$visMUAC)
#0.775
pairwise.wilcox.test(metadatatime12$visWHZ, metadatatime12$Treatment)
#p=0.46
median(time12amox$visWHZ)
#-0.66
IQR(time12amox$visWHZ)
#1.27
median(time12placebo$visWHZ)
#-1.065
IQR(time12placebo$visWHZ)
#1.0125
pairwise.wilcox.test(as.numeric(metadatatime12$visdelMUAC), metadatatime12$Treatment)
#p=0.18
median(time12amox$visdelMUAC)
#1.3
IQR(time12amox$visdelMUAC)
#1
median(time12placebo$visdelMUAC)
#1.1
IQR(time12placebo$visdelMUAC)
#0.75
pairwise.wilcox.test(metadatatime12$visdelWHZ, metadatatime12$Treatment)
#p=0.093
median(time12amox$visdelWHZ)
#2.28
IQR(time12amox$visdelWHZ)
#1.41
median(time12placebo$visdelWHZ)
#1.965
IQR(time12placebo$visdelWHZ)
#1.1425

pairwise.wilcox.test(metadatatime104$SAGE, metadatatime104$Treatment, p.adjust.method = "BH")
#             Amoxicillin Healthy Placebo
#Healthy    1.5e-06     -       -      
#Placebo    0.57362     4.4e-05 -      
#SAMcontrol 0.00052     0.57362 0.00131

median(time104amox$SAGE, na.rm = TRUE)
#36
IQR(time104amox$SAGE, na.rm = TRUE)
#24
median(time104placebo$SAGE,na.rm = TRUE)
#38.5
IQR(time104placebo$SAGE, na.rm = TRUE)
#9.25
median(time104healthy$SAGE)
#12
IQR(time104healthy$SAGE)
#13
median(time104SAM$SAGE)
#10.5
IQR(time104SAM$SAGE)
#5.5
table(time104amox$SSEX)
#10M and 11F (48%M)
table(time104placebo$SSEX)
#8M and 4F (67%M)
table(time104healthy$SSEX)
#21M and 17F (55%M)
table(time104SAM$SSEX)
#3M and 3F (50%M)
chisq.test(metadatatime104$SSEX, metadatatime104$Treatment)
#p=0.7589
#Just amoxicillin and placebo for delWHZ and delMUAC
metadatatime104amoxpla<-subset(metadatatime104, Treatment=="Amoxicillin"|Treatment=="Placebo")
pairwise.wilcox.test(as.numeric(metadatatime104amoxpla$SAGE), metadatatime104amoxpla$Treatment)
#p=0.57
pairwise.wilcox.test(as.numeric(metadatatime104amoxpla$visdelMUAC), metadatatime104amoxpla$Treatment)
#p=0.44
median(time104amox$visdelMUAC, na.rm = TRUE)
#3.7
IQR(time104amox$visdelMUAC,na.rm = TRUE)
#1.4
median(time104placebo$visdelMUAC,na.rm = TRUE)
#3.45
IQR(time104placebo$visdelMUAC, na.rm = TRUE)
#1.375
pairwise.wilcox.test(metadatatime104amoxpla$visdelWHZ, metadatatime104amoxpla$Treatment)
#p=0.38
median(time104amox$visdelWHZ, na.rm = TRUE)
#3.29
IQR(time104amox$visdelWHZ, na.rm = TRUE)
#1.49
median(time104placebo$visdelWHZ, na.rm = TRUE)
#3.08
IQR(time104placebo$visdelWHZ, na.rm = TRUE)
#1.6175

pairwise.wilcox.test(metadatatime104$SMUAC, metadatatime104$Treatment, p.adjust.method = "BH")
#           Amoxicillin Healthy Placebo
#Healthy    0.05767     -       -      
#Placebo    0.25826     0.51594 -      
#SAMcontrol 0.00079     0.00060 0.00172

median(time104amox$SMUAC, na.rm = TRUE)
#14.6
IQR(time104amox$SMUAC, na.rm = TRUE)
#1.4
median(time104placebo$SMUAC, na.rm = TRUE)
#14
IQR(time104placebo$SMUAC, na.rm = TRUE)
#0.9
median(time104healthy$SMUAC)
#13.9
IQR(time104healthy$SMUAC)
#1.35
median(time104SAM$SMUAC)
#11.65
IQR(time104SAM$SMUAC)
#0.625

pairwise.wilcox.test(metadatatime104$SZWFL, metadatatime104$Treatment, p.adjust.method = "BH")
#           Amoxicillin Healthy Placebo
# Healthy    0.09422     -       -      
# Placebo    0.62521     0.62521 -      
# SAMcontrol 0.00053     0.00032 0.00032

median(time104amox$SZWFL, na.rm = TRUE)
#-0.21
IQR(time104amox$SZWFL, na.rm = TRUE)
#1.8
median(time104placebo$SZWFL, na.rm = TRUE)
#-0.23
IQR(time104placebo$SZWFL, na.rm = TRUE)
#1.415
median(time104healthy$SZWFL, na.rm = TRUE)
#-0.99
IQR(time104healthy$SZWFL, na.rm = TRUE)
#1.845
median(time104SAM$SZWFL, na.rm = TRUE)
#-3.23
IQR(time104SAM$SZWFL, na.rm = TRUE)
#0.16

wilcox.test(as.numeric(metadatatime104$delZWFL)~metadatatime104$Treatment, p.adjust.method = "BH",na.rm = TRUE)
median(time104amox$delZWFL, na.rm = TRUE)
median(time104placebo$delZWFL, na.rm = TRUE)

wilcox.test(as.numeric(metadatatime104$delMUAC)~metadatatime104$Treatment, p.adjust.method = "BH",na.rm = TRUE)

ggplot(metadatatime0, aes(x=as.factor(Week_of_follow_up), y=BAGE, color=Treatment))+ geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Followup Age")+xlab("Timepoint")+ylab("Age") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/211216_Ageatbaseline.pdf")

ggplot(metadatatime104, aes(x=as.factor(Week_of_follow_up), y=SAGE, color=Treatment))+ geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Followup Age")+xlab("Timepoint")+ylab("Age") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/211216_Ageatfollowup.pdf")

#Readcount at each timepoint
ggplot(metadata, aes(x=as.factor(Week_of_follow_up), y=Post.processed_readcount, color=Treatment)) +geom_boxplot(outlier.shape = NA)+geom_point(position = position_jitterdodge(jitter.width = 0.1))+xlab("Timepoint")+ylab("ReadCount") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/211216_Readcount.pdf")

#Any statistical differences in readcount at timepoint 104?
pairwise.wilcox.test(metadata$Post.processed_readcount, metadata$Treatment, p.adjust.method = "none")
#           Amoxicillin Healthy Placebo
#Healthy    0.872       -       -      
#Placebo    0.053       0.349   -      
#SAMcontrol 0.637       0.536   0.960  
#P value adjustment method: none

#LME of WHZ, MUAC, and delta over time of just placebo and amox
treatment<-subset(metadata, Treatment=="Amoxicillin" | Treatment=="Placebo")
visMUAClme<-lmer(visMUAC~Treatment+Week_of_follow_up + (1|idno), data=treatment)
anova(visMUAClme)
#Type III Analysis of Variance Table with Satterthwaite's method
#                   Sum Sq Mean Sq NumDF  DenDF  F value  Pr(>F)    
#Treatment           2.004   2.004     1 145.52   5.1579 0.02461 *  
#Week_of_follow_up 266.658 266.658     1 263.80 686.4694 < 2e-16 ***
#---
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


ggplot(treatment, aes(x=as.factor(Week_of_follow_up), y=visMUAC, color=Treatment))+geom_boxplot()+geom_smooth(method="lm",formula=y~x, se=TRUE)+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220719_visMUAC.pdf")

WHZlme<-lmer(visWHZ~Treatment+Week_of_follow_up + (1|idno), data=treatment)
anova(WHZlme)
#Type III Analysis of Variance Table with Satterthwaite's method
#                   Sum Sq Mean Sq NumDF  DenDF F value Pr(>F)    
# Treatment           0.968   0.968     1 138.40   0.9881 0.3219    
# Week_of_follow_up 115.299 115.299     1 291.09 117.7056 <2e-16 *** 

ggplot(treatment, aes(x=as.factor(Week_of_follow_up), y=visWHZ, color=Treatment))+geom_boxplot()+geom_smooth(method="lm",formula=y~x, se=TRUE)+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/221122_visWHZ.pdf")

visdelMUAClme<-lmer(visdelMUAC~Treatment+Week_of_follow_up + (1|idno), data=treatment)
anova(visdelMUAClme)
#Type III Analysis of Variance Table with Satterthwaite's method
#                   Sum Sq Mean Sq NumDF  DenDF  F value  Pr(>F)    
#Treatment           1.447   1.447     1 146.35   4.3907 0.03786 *  
#Week_of_follow_up 208.953 208.953     1 253.31 634.1916 < 2e-16 ***
#---
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

ggplot(treatment, aes(x=as.factor(Week_of_follow_up), y=visdelMUAC, color=Treatment))+geom_boxplot()+geom_smooth(method="lm",formula=y~x, se=TRUE)+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220719_visdelMUAC.pdf")

visdelWHZlme<-lmer(visdelWHZ~Treatment+Week_of_follow_up + (1|idno), data=treatment)
anova(visdelWHZlme)
#Type III Analysis of Variance Table with Satterthwaite's method
#                  Sum Sq Mean Sq NumDF  DenDF F value    Pr(>F)    
# Treatment          5.172   5.172     1 131.86  7.7856  0.006048 ** 
# Week_of_follow_up 58.031  58.031     1 242.74 87.3558 < 2.2e-16 ***


ggplot(treatment, aes(x=as.factor(Week_of_follow_up), y=visdelWHZ, color=Treatment))+geom_boxplot()+geom_smooth(method="lm",formula=y~x, se=TRUE)+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/221122_visdelWHZ.pdf")
```

```{r}
#Bray-curtis distance between microbiomes
Specieswidemeta<-join(metadata, Specieswide)
Week104speciesmeta<-subset(Specieswidemeta, Week_of_follow_up==104)
.rowNamesDF(Specieswidemeta)<-Specieswidemeta$Sample
Specieswidemeta<-Specieswidemeta[-c(1:212)]

Bray <-vegdist(Specieswidemeta, method = "bray")
pco.brayDF<-pco(Bray, k=4)
BrayDF<-as.data.frame(pco.brayDF$points)

BrayDF$Sample<-row.names(BrayDF)
BrayDFmeta<-join(BrayDF, metadata, by = "Sample")
### Eigenvalues for all data##
EigenBray<- rbind(SD = sqrt(pco.brayDF$eig),Proportion = pco.brayDF$eig/sum(pco.brayDF$eig),Cumulative = cumsum(pco.brayDF$eig)/sum(pco.brayDF$eig))
#V1 = 23.5 and V2 = 11.3
BrayDF$Sample=NULL

ggplot(BrayDFmeta, aes(x=V1, y=V2, color=Treatment, shape=as.factor(Week_of_follow_up))) +geom_point(size=5)+
  theme_bw()+theme(legend.text=element_text(size=12, face="bold"),legend.title=element_text(size=14, face="bold"),
        panel.grid.minor = element_blank(), panel.border = element_rect(colour = "black"), 
        axis.text.x= element_text(size=12,angle=45,hjust=1,vjust=1,face="bold"),
        axis.text.y= element_text(size=12,face="bold"))+ scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))+
  ylab("PCO2 (11.3%)")+
  xlab("PCO1 (23.5%)")

#Week 0 and 1 only
Specieswidemeta<-join(metadata, Specieswide)
Week0to1speciesmeta<-subset(Specieswidemeta, Week_of_follow_up==0|Week_of_follow_up==1)
.rowNamesDF(Week0to1speciesmeta)<-Week0to1speciesmeta$Sample
Week0to1Bray<-Week0to1speciesmeta[-c(1:212)]
Bray0to1 <-vegdist(Week0to1Bray, method = "bray")
pco.brayDF0to1<-pco(Bray0to1, k=4)
Bray0to1<-as.data.frame(pco.brayDF0to1$points)
Bray0to1$Sample<-row.names(Bray0to1)
BrayDF0to1meta<-join(Bray0to1, metadata, by = "Sample")
Bray0to1$Sample=NULL
### Eigenvalues for all data##
EigenBray0to1<- rbind(SD = sqrt(pco.brayDF0to1$eig),Proportion = pco.brayDF0to1$eig/sum(pco.brayDF0to1$eig),Cumulative = cumsum(pco.brayDF0to1$eig)/sum(pco.brayDF0to1$eig))
#V1 = 24.2 and V2 = 16.8
BrayDF0to1meta$Sample=NULL

BrayDF0to1meta$Description=factor(BrayDF0to1meta$Description, levels = c("SAM_Placebo_wk0","SAM_Amoxicillin_wk0","SAM_Placebo_wk1","SAM_Amoxicillin_wk1"))

q<-ggplot(BrayDF0to1meta, aes(x=V1, y=V2, color=Description, shape=as.factor(Week_of_follow_up))) +geom_point(size=5)+
  theme_bw()+theme(legend.text=element_text(size=12, face="bold"),legend.title=element_text(size=14, face="bold"),
        panel.grid.minor = element_blank(), panel.border = element_rect(colour = "black"), 
        axis.text.x= element_text(size=12,angle=45,hjust=1,vjust=1,face="bold"),
        axis.text.y= element_text(size=12,face="bold"))+stat_ellipse(geom='path', position = 'identity',type = 't', level=0.95, inherit.aes=TRUE)+
  ylab("PCO2 (16.8%)")+
  xlab("PCO1 (24.2%)")+scale_color_manual(values = c("SAM_Placebo_wk0" = "#1F78B4","SAM_Amoxicillin_wk0" = "#E31A1C","SAM_Placebo_wk1" = "#0000CC","SAM_Amoxicillin_wk1" = "#990000"))
ggMarginal(q, type="boxplot", groupColour = TRUE, size=5)
ggsave(file="2021Figs/220721_Week0to1BrayPCOA.pdf")

adonis2(Bray0to1~Treatment, data = BrayDF0to1meta, permutations = 999, method ="bray")
#            Df SumOfSqs       R2       F Pr(>F)
# Treatment   1   -36162 -0.06538 -8.4078  0.549
# Residual  137   589238  1.06538               
# Total     138   553076  1.00000
pairwise.wilcox.test(Bray0to1$V1, BrayDF0to1meta$Description, p.adjust.method = "BH")
#                     SAM_Amoxicillin_wk0 SAM_Amoxicillin_wk1 SAM_Placebo_wk0
# SAM_Amoxicillin_wk1 0.63                -                   -              
# SAM_Placebo_wk0     1.00                0.63                -              
# SAM_Placebo_wk1     0.63                1.00                0.63           
pairwise.wilcox.test(Bray0to1$V2, BrayDF0to1meta$Description, p.adjust.method = "BH")
#                     SAM_Amoxicillin_wk0 SAM_Amoxicillin_wk1 SAM_Placebo_wk0
# SAM_Amoxicillin_wk1 0.01216             -                   -              
# SAM_Placebo_wk0     0.30939             0.00531             -              
# SAM_Placebo_wk1     0.26736             0.00082             0.94287        

#ggplot(BrayDF0to1meta, aes(x=Treatment, y=V1, color=Treatment))+geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("BC #V1")+xlab("Treatment")+ylab("PCoA V1") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = #element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = #"#FF7F00", "Healthy" = "#33A02C"))


#What about just at the week 104 timepoint?
Week104speciesmeta<-subset(Specieswidemeta, Week_of_follow_up==104)
.rowNamesDF(Week104speciesmeta)<-Week104speciesmeta$Sample
Week104Bray<-Week104speciesmeta[-c(1:212)]
Bray104 <-vegdist(Week104Bray, method = "bray")
pco.brayDF104<-pco(Bray104, k=4)
Bray104<-as.data.frame(pco.brayDF104$points)
### Eigenvalues for all data##
EigenBray104<- rbind(SD = sqrt(pco.brayDF104$eig),Proportion = pco.brayDF104$eig/sum(pco.brayDF104$eig),Cumulative = cumsum(pco.brayDF104$eig)/sum(pco.brayDF104$eig))
#V1 = 27.8 and V2 = 9.4
BrayDF$Sample=NULL

Bray104$Sample<-rownames(Bray104)
Bray104meta<-join(Bray104, metadata)
Bray104$Sample=NULL

p<-ggplot(Bray104meta, aes(x=V1, y=V2, color=Treatment)) +geom_point(size=5)+
  theme_bw()+theme(legend.text=element_text(size=12, face="bold"),legend.title=element_text(size=14, face="bold"),
        panel.grid.minor = element_blank(), panel.border = element_rect(colour = "black"), 
        axis.text.x= element_text(size=12,angle=45,hjust=1,vjust=1,face="bold"),
        axis.text.y= element_text(size=12,face="bold"))+ scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))+stat_ellipse(geom='path', position = 'identity',type = 't', level=0.95)+
  ylab("PCO2 (9.4%)")+
  xlab("PCO1 (27.8%)")
ggsave(file="2021Figs/220608_Week104BrayPCOA.pdf")


ggMarginal(p, type="boxplot", groupColour = TRUE, size=5, margins = "x")
ggsave(file="2021Figs/220630_Week104BrayPCOAwboxplot.pdf")

adonis2(Bray104~Treatment, data = Bray104meta, permutations = 999, method = "bray") 
 
#           Df SumOfSqs      R2      F Pr(>F)
# Treatment  3    17736 0.35094 13.517  0.545
# Residual  75    32801 0.64906              
# Total     78    50537 1.00000       

ggplot(Bray104meta, aes(x=Treatment, y=V1, color=Treatment))+geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("BC V1")+xlab("Treatment")+ylab("PCoA V1") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Week104BrayPCOAV1.pdf")

kruskal.test(V1~Treatment, data = Bray104meta)
#Kruskal-Wallis chi-squared = 13.04, df = 3, p-value = 0.004551
pairwise.wilcox.test(Bray104meta$V1, Bray104meta$Treatment, p.adjust.method = "BH")
#            Amoxicillin Healthy Placebo
# Healthy    0.0064      -       -      
# Placebo    0.1916      0.4223  -      
# SAMcontrol 0.0356      0.1916  0.1916 
# 
# P value adjustment method: BH 

ggplot(Bray104meta, aes(x=Treatment, y=V2, color=Treatment))+geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("BC V1")+xlab("Treatment")+ylab("PCoA V1") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Week104BrayPCOAV2.pdf")

kruskal.test(V2~Treatment, data = Bray104meta)
# Kruskal-Wallis chi-squared = 0.63556, df = 3, p-value = 0.8882
pairwise.wilcox.test(Bray104meta$V2, Bray104meta$Treatment, p.adjust.method = "BH")
#            Amoxicillin Healthy Placebo
# Healthy    0.99        -       -      
# Placebo    0.99        0.99    -      
# SAMcontrol 0.99        0.99    0.99  

#Bray-Curtis DC of just week 0 and week 1 data


```


```{r}
#ARG analysis
Abx_geneslong<-read.delim("Shortbred_data/Data_tables/220526_Nigershortbred.txt")
#Generate wide version
# ABX_wide<-Abx_geneslong[-c(4:18)]
# ABX_wide<-reshape(ABX_wide, idvar="Sample", timevar = "Family", direction ="wide")
# ABX_wide[is.na(ABX_wide)] <- 0
# rownames(ABX_wide)=ABX_wide$Sample
# ABX_wide$Sample=NULL
# ABX_wide$Sum<-rowSums(ABX_wide)
# sample_df = data.frame(samples=unique(ABX_wide$Sample), count=sapply(unique(ABX_wide$Sample), function(s) dim(ABX_wide[ABX_wide$Sample == s,])[1]))
# ABX_wide$Sample<-row.names(ABX_wide)
# write.csv(ABX_wide,"Shortbred_data/Data_tables/200805_Shortbredwide_RPKM.csv")
ABX_wide<-read.delim("Shortbred_data/Data_tables/200526_Shortbredwide_RPKM.csv", sep=",")

# #Wide version with shortbred ID
# ABX_wideID<-Abx_geneslong[-c(4:5,7:18)]
# ABX_wideID<-ABX_wideID[-c(1)]
# ABX_wideID<-reshape(ABX_wideID, idvar="Sample", timevar = "Shortbred_ID", direction ="wide")
# ABX_wideID[is.na(ABX_wideID)] <- 0
# rownames(ABX_wideID)=ABX_wideID$Sample
# ABX_wideID$Sample=NULL
# ABX_wideID$Sum<-rowSums(ABX_wideID)
# sample_df = data.frame(samples=unique(ABX_wide$Sample), count=sapply(unique(ABX_wide$Sample), function(s) dim(ABX_wide[ABX_wide$Sample == s,])[1]))
# ABX_wide$Sample<-row.names(ABX_wide)
# write.csv(ABX_wideID,"Shortbred_data/Data_tables/220818_Shortbredwide_wide_ShortbredID.csv")

Shortbredmetadata<-join(metadata,ABX_wide, by ="Sample")
Shortbredmetadata$Treatment=factor(Shortbredmetadata$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
ARGsummary<-aggregate(Sum ~ Description,Shortbredmetadata, FUN=median)
ARGsummary$medianRPKM<-aggregate(Sum ~ Description,Shortbredmetadata, FUN=median)
ARGsummary$RPKMIQR<-aggregate(Sum ~ Description,Shortbredmetadata, FUN=IQR)
ARGsummary$countmedian<-aggregate(Count ~ Description,Shortbredmetadata, FUN=median)
ARGsummary$countIQR<-aggregate(Count ~ Description,Shortbredmetadata, FUN=IQR)
write.csv(ARGsummary, "Shortbredsummarystatistics.csv")


#Plotting RPKM sum and then unique count
ggplot(Shortbredmetadata, aes(x=as.factor(Week_of_follow_up), y=Sum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Total ARGs")+xlab("Timepoint")+ylab("Log total ARGs/sample") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))+scale_y_continuous(trans = "log10")
ggsave(file="2021Figs/220714_ARGsumovertime.pdf")

ggplot(Shortbredmetadata, aes(x=as.factor(Week_of_follow_up), y=Count, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Unique ARGs")+xlab("Timepoint")+ylab("Unique ARGs") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220714_UniqueARGCountovertime.pdf")

fitARG<- lmer(Sum~Description+(1|idno), data=Shortbredmetadata)
summary(fitARG)
TukfitfitARG<-lsmeans(fitARG, pairwise~Description, adjust="Tukey")
TukfitfitARG
#only identified significant difference after correction
# contrast                                     estimate   SE  df t.ratio p.value
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104    3790.0 1021 292   3.712  0.0172
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4      3008.4  900 317   3.341  0.0563
# SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8      3134.4  929 296   3.373  0.0513

fitARGcount<- lmer(Count~Description+(1|idno), data=Shortbredmetadata)
summary(fitARGcount)
TukfitARGcount<-lsmeans(fitARGcount, pairwise~Description, adjust="Tukey")
TukfitARGcount
#Signficant or near-significant differences for Count
#contrast                                     estimate    SE  df t.ratio p.value
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk0        -11.229  6.18 414  -1.817  0.8674
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk1        -25.001  6.01 414  -4.159  0.0031
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk104        8.733  7.08 414   1.234  0.9942
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk12        -5.191  5.94 414  -0.874  0.9998
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk4          3.906  6.18 414   0.632  1.0000
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk8          3.462  6.45 414   0.537  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk0              2.895  6.33 414   0.457  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk1             -0.101  6.28 414  -0.016  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk104           -1.306  8.49 414  -0.154  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk12            -0.422  6.23 414  -0.068  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk4              3.202  5.87 414   0.545  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk8              1.591  6.64 414   0.239  1.0000
 # Healthy_NA_wk104 - SAMcontrol_NA_wk104        -20.105 11.57 414  -1.737  0.9015
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk1     -13.773  6.12 334  -2.250  0.5934
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk104    19.961  7.16 325   2.789  0.2356
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk12      6.038  6.06 364   0.997  0.9993
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk4      15.135  6.29 351   2.406  0.4782
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk8      14.691  6.55 348   2.243  0.5982
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk0          14.123  6.46 414   2.187  0.6392
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk1          11.128  6.41 414   1.737  0.9016
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk104         9.922  8.59 414   1.155  0.9969
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk12         10.806  6.36 414   1.700  0.9153
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk4          14.430  6.01 414   2.401  0.4810
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk8          12.820  6.77 414   1.895  0.8285
 # SAM_Amoxicillin_wk0 - SAMcontrol_NA_wk104      -8.877 11.65 414  -0.762  1.0000
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104    33.734  7.03 362   4.801  0.0002
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk12     19.810  5.89 372   3.363  0.0521
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4      28.907  6.13 361   4.716  0.0003
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8      28.464  6.39 354   4.452  0.0009
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk0          27.896  6.30 414   4.430  0.0010
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk1          24.901  6.24 414   3.987  0.0060
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk104        23.695  8.47 414   2.798  0.2295
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk12         24.579  6.19 414   3.968  0.0065
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk4          28.203  5.84 414   4.832  0.0002
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk8          26.592  6.61 414   4.021  0.0053
 # SAM_Amoxicillin_wk1 - SAMcontrol_NA_wk104       4.896 11.56 414   0.424  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk12  -13.924  6.97 374  -1.998  0.7680
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk4    -4.827  7.17 372  -0.673  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk8    -5.271  7.40 364  -0.712  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk0        -5.838  7.32 414  -0.797  0.9999
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk1        -8.833  7.28 414  -1.214  0.9950
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk104     -10.039  9.25 414  -1.085  0.9984
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk12       -9.155  7.23 414  -1.266  0.9926
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk4        -5.531  6.93 414  -0.798  0.9999
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk8        -7.142  7.59 414  -0.940  0.9996
 # SAM_Amoxicillin_wk104 - SAMcontrol_NA_wk104   -28.838 12.15 414  -2.374  0.5008
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk4      9.097  6.06 364   1.501  0.9668
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk8      8.653  6.32 338   1.369  0.9847
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk0          8.086  6.23 414   1.298  0.9906
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk1          5.090  6.17 414   0.824  0.9999
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk104        3.885  8.42 414   0.462  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk12         4.769  6.12 414   0.779  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk4          8.393  5.76 414   1.457  0.9742
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk8          6.782  6.55 414   1.036  0.9990
 # SAM_Amoxicillin_wk12 - SAMcontrol_NA_wk104    -14.914 11.52 414  -1.295  0.9908
 # SAM_Amoxicillin_wk4 - SAM_Amoxicillin_wk8      -0.444  6.55 342  -0.068  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk0          -1.011  6.46 414  -0.157  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk1          -4.007  6.41 414  -0.625  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk104        -5.212  8.59 414  -0.607  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk12         -4.328  6.36 414  -0.681  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk4          -0.704  6.01 414  -0.117  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk8          -2.315  6.77 414  -0.342  1.0000
 # SAM_Amoxicillin_wk4 - SAMcontrol_NA_wk104     -24.011 11.65 414  -2.062  0.7270
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk0          -0.567  6.71 414  -0.085  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk1          -3.563  6.66 414  -0.535  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk104        -4.769  8.78 414  -0.543  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk12         -3.885  6.62 414  -0.587  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk4          -0.260  6.28 414  -0.041  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk8          -1.871  7.01 414  -0.267  1.0000
 # SAM_Amoxicillin_wk8 - SAMcontrol_NA_wk104     -23.567 11.79 414  -1.999  0.7676
 # SAM_Placebo_wk0 - SAM_Placebo_wk1              -2.995  6.53 359  -0.459  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk104            -4.201  8.67 358  -0.485  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk12             -3.317  6.49 394  -0.511  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk4               0.307  6.15 386   0.050  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk8              -1.304  6.89 389  -0.189  1.0000
 # SAM_Placebo_wk0 - SAMcontrol_NA_wk104         -23.000 11.73 414  -1.962  0.7906
 # SAM_Placebo_wk1 - SAM_Placebo_wk104            -1.206  8.64 391  -0.139  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk12             -0.322  6.44 384  -0.050  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk4               3.302  6.10 387   0.541  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk8               1.692  6.84 377   0.247  1.0000
 # SAM_Placebo_wk1 - SAMcontrol_NA_wk104         -20.005 11.70 414  -1.710  0.9116
 # SAM_Placebo_wk104 - SAM_Placebo_wk12            0.884  8.62 408   0.103  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk4             4.508  8.36 405   0.539  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk8             2.897  8.92 406   0.325  1.0000
 # SAM_Placebo_wk104 - SAMcontrol_NA_wk104       -18.799 13.02 414  -1.444  0.9761
 # SAM_Placebo_wk12 - SAM_Placebo_wk4              3.624  6.04 377   0.600  1.0000
 # SAM_Placebo_wk12 - SAM_Placebo_wk8              2.013  6.79 363   0.297  1.0000
 # SAM_Placebo_wk12 - SAMcontrol_NA_wk104        -19.683 11.67 414  -1.687  0.9198
 # SAM_Placebo_wk4 - SAM_Placebo_wk8              -1.611  6.46 361  -0.249  1.0000
 # SAM_Placebo_wk4 - SAMcontrol_NA_wk104         -23.307 11.48 414  -2.029  0.7483
 # SAM_Placebo_wk8 - SAMcontrol_NA_wk104         -21.696 11.90 414  -1.824  0.8644

#Now to evaluate only beta lactamases
# Betalactamases<-Abx_geneslong %>% filter(grepl('Beta-lactamase',AMR_Classification))
# Betalactamases<-Betalactamases[-c(4:18)]
# Betalactamases<-reshape(Betalactamases, idvar="Sample", timevar = "Family", direction ="wide")
# Betalactamases[is.na(Betalactamases)] <- 0
# rownames(Betalactamases)=Betalactamases$Sample
# Betalactamases$Sample=NULL
# Betalactamases$Sum<-rowSums(Betalactamases)
# Betalactamases$Sample<-row.names(Betalactamases)
# write.csv(Betalactamases,"Shortbred_data/Data_tables/220620_betalactamases_RPKM.csv")
Betalactamases<-read.delim("Shortbred_data/Data_tables/220620_betalactamases_RPKM.csv", sep=",")
metabeta<-join(metadata, Betalactamases)
write.csv(metabeta,"Shortbred_data/Data_tables/221109_betalactamases_RPKM_metadata.csv")
#Adding zeroes to those without Beta lactamases
metabeta$Count[is.na(metabeta$Count)] <- 0
metabeta$Sum[is.na(metabeta$Sum)] <- 0
metabeta$Treatment=factor(metabeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
ggplot(metabeta, aes(x=as.factor(Week_of_follow_up), y=Sum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Beta Lactamase RPKM")+xlab("Timepoint")+ylab("Beta lactamases") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))+scale_y_continuous(trans = "log10")
ggsave(file="2021Figs/221109_BetalactamaseRPKMovertimewithzeroes.pdf")

ggplot(metabeta, aes(x=as.factor(Week_of_follow_up), y=Count, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Beta Lactamase count per sample")+xlab("Timepoint")+ylab("Beta lactamases") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/221109_Betalactamasecountovertimewithzeroes.pdf")

fitbetasum<- lmer(Sum~Description+(1|idno), data=metabeta)
summary(fitbetasum)
Tukfitbetasum<-lsmeans(fitbetasum, pairwise~Description, adjust="Tukey")
Tukfitbetasum
 # contrast                                     estimate  SE  df t.ratio p.value
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk0        -304.00 229 381  -1.329  0.9883
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk1        -893.59 223 373  -3.999  0.0058
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk104       100.16 255 409   0.392  1.0000
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk12       -218.50 221 369  -0.988  0.9994
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk4        -105.72 229 381  -0.463  1.0000
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk8        -136.43 236 394  -0.577  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk0            -358.74 234 387  -1.535  0.9604
 # Healthy_NA_wk104 - SAM_Placebo_wk1            -147.95 232 387  -0.639  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk104           -44.38 301 412  -0.147  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk12            -46.59 230 383  -0.202  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk4             -12.90 219 370  -0.059  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk8            -224.38 243 401  -0.925  0.9997
 # Healthy_NA_wk104 - SAMcontrol_NA_wk104        -439.50 439 330  -1.001  0.9993
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk1     -589.58 193 279  -3.049  0.1278
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk104    404.16 222 250   1.824  0.8632
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk12      85.51 196 308   0.435  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk4      198.28 201 291   0.986  0.9994
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk8      167.57 207 274   0.808  0.9999
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk0          -54.74 233 413  -0.235  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk1          156.06 231 413   0.675  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk104        259.63 301 399   0.864  0.9999
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk12         257.42 230 412   1.121  0.9977
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk4          291.11 218 409   1.335  0.9879
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk8           79.62 242 414   0.329  1.0000
 # SAM_Amoxicillin_wk0 - SAMcontrol_NA_wk104     -135.50 439 345  -0.309  1.0000
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104    993.75 224 282   4.433  0.0011
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk12     675.09 193 323   3.500  0.0344
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4      787.86 198 308   3.971  0.0067
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8      757.15 204 286   3.709  0.0174
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk0          534.84 228 411   2.347  0.5213
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk1          745.64 226 411   3.302  0.0621
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk104        849.21 297 401   2.863  0.1984
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk12         847.00 224 410   3.775  0.0132
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk4          880.69 213 405   4.143  0.0033
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk8          669.21 237 414   2.823  0.2170
 # SAM_Amoxicillin_wk1 - SAMcontrol_NA_wk104      454.08 436 342   1.041  0.9989
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk12  -318.66 225 296  -1.415  0.9796
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk4   -205.89 230 287  -0.893  0.9998
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk8   -236.59 235 272  -1.006  0.9992
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk0       -458.90 259 411  -1.771  0.8879
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk1       -248.11 257 411  -0.964  0.9995
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk104     -144.54 321 389  -0.450  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk12      -146.75 256 412  -0.573  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk4       -113.06 246 413  -0.460  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk8       -324.54 267 407  -1.215  0.9950
 # SAM_Amoxicillin_wk104 - SAMcontrol_NA_wk104   -539.66 453 362  -1.191  0.9958
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk4     112.77 197 314   0.573  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk8      82.06 200 279   0.411  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk0        -140.25 226 410  -0.622  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk1          70.55 223 410   0.316  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk104       174.12 295 401   0.591  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk12        171.91 222 409   0.774  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk4         205.60 210 404   0.978  0.9994
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk8          -5.88 235 414  -0.025  1.0000
 # SAM_Amoxicillin_wk12 - SAMcontrol_NA_wk104    -221.00 435 340  -0.508  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Amoxicillin_wk8      -30.71 207 276  -0.148  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk0         -253.02 233 413  -1.086  0.9983
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk1          -42.22 231 413  -0.183  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk104         61.35 300 399   0.204  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk12          59.14 229 412   0.258  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk4           92.83 218 409   0.426  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk8         -118.66 242 414  -0.491  1.0000
 # SAM_Amoxicillin_wk4 - SAMcontrol_NA_wk104     -333.78 439 345  -0.761  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk0         -222.31 240 414  -0.924  0.9997
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk1          -11.51 238 414  -0.048  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk104         92.06 306 396   0.300  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk12          89.85 237 414   0.379  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk4          123.54 226 413   0.547  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk8          -87.95 249 413  -0.353  1.0000
 # SAM_Amoxicillin_wk8 - SAMcontrol_NA_wk104     -303.07 443 350  -0.684  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk1              210.80 211 312   0.997  0.9993
 # SAM_Placebo_wk0 - SAM_Placebo_wk104            314.37 275 274   1.143  0.9972
 # SAM_Placebo_wk0 - SAM_Placebo_wk12             312.16 219 359   1.426  0.9783
 # SAM_Placebo_wk0 - SAM_Placebo_wk4              345.85 205 346   1.686  0.9197
 # SAM_Placebo_wk0 - SAM_Placebo_wk8              134.36 229 334   0.587  1.0000
 # SAM_Placebo_wk0 - SAMcontrol_NA_wk104          -80.76 441 347  -0.183  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk104            103.57 283 304   0.366  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk12             101.36 215 349   0.472  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk4              135.05 204 349   0.663  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk8              -76.43 225 324  -0.340  1.0000
 # SAM_Placebo_wk1 - SAMcontrol_NA_wk104         -291.56 440 347  -0.662  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk12            -2.21 290 337  -0.008  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk4             31.48 279 327   0.113  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk8           -180.00 297 322  -0.606  1.0000
 # SAM_Placebo_wk104 - SAMcontrol_NA_wk104       -395.13 481 384  -0.822  0.9999
 # SAM_Placebo_wk12 - SAM_Placebo_wk4              33.69 200 342   0.168  1.0000
 # SAM_Placebo_wk12 - SAM_Placebo_wk8            -177.79 220 313  -0.807  0.9999
 # SAM_Placebo_wk12 - SAMcontrol_NA_wk104        -392.91 440 346  -0.894  0.9998
 # SAM_Placebo_wk4 - SAM_Placebo_wk8             -211.48 209 305  -1.013  0.9992
 # SAM_Placebo_wk4 - SAMcontrol_NA_wk104         -426.60 434 340  -0.984  0.9994
 # SAM_Placebo_wk8 - SAMcontrol_NA_wk104         -215.12 446 354  -0.482  1.0000

fitbetacount<- lmer(Count~Description+(1|idno), data=metabeta)
summary(fitbetacount)
tukfitbetacount<-lsmeans(fitbetacount, pairwise~Description, adjust="Tukey")
tukfitbetacount
 # contrast                                     estimate    SE  df t.ratio p.value
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk0       -1.05556 0.851 412  -1.240  0.9939
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk1       -3.63436 0.828 412  -4.389  0.0012
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk104      0.66202 0.972 414   0.681  1.0000
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk12      -1.06533 0.818 412  -1.302  0.9903
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk4        0.02332 0.851 412   0.027  1.0000
 # Healthy_NA_wk104 - SAM_Amoxicillin_wk8       -0.96239 0.887 413  -1.085  0.9984
 # Healthy_NA_wk104 - SAM_Placebo_wk0           -0.25484 0.872 412  -0.292  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk1            0.10401 0.864 413   0.120  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk104         -0.20467 1.166 414  -0.176  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk12          -0.06232 0.858 412  -0.073  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk4            0.59126 0.809 411   0.731  1.0000
 # Healthy_NA_wk104 - SAM_Placebo_wk8            0.19214 0.914 414   0.210  1.0000
 # Healthy_NA_wk104 - SAMcontrol_NA_wk104       -2.64035 1.596 408  -1.654  0.9303
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk1    -2.57881 0.821 321  -3.143  0.0990
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk104   1.71757 0.957 302   1.796  0.8767
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk12   -0.00978 0.817 353  -0.012  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk4     1.07887 0.846 337   1.275  0.9920
 # SAM_Amoxicillin_wk0 - SAM_Amoxicillin_wk8     0.09316 0.880 330   0.106  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk0         0.80072 0.888 414   0.901  0.9998
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk1         1.15956 0.881 414   1.316  0.9894
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk104       0.85089 1.179 413   0.722  1.0000
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk12        0.99324 0.874 414   1.136  0.9974
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk4         1.64682 0.827 414   1.992  0.7722
 # SAM_Amoxicillin_wk0 - SAM_Placebo_wk8         1.24770 0.930 414   1.342  0.9874
 # SAM_Amoxicillin_wk0 - SAMcontrol_NA_wk104    -1.58479 1.606 409  -0.987  0.9994
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk104   4.29638 0.946 342   4.542  0.0006
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk12    2.56903 0.796 363   3.228  0.0776
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk4     3.65768 0.826 350   4.428  0.0010
 # SAM_Amoxicillin_wk1 - SAM_Amoxicillin_wk8     2.67197 0.860 339   3.107  0.1087
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk0         3.37953 0.866 414   3.902  0.0083
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk1         3.73837 0.859 414   4.352  0.0014
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk104       3.42969 1.162 413   2.951  0.1608
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk12        3.57205 0.852 414   4.192  0.0027
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk4         4.22562 0.803 414   5.262  <.0001
 # SAM_Amoxicillin_wk1 - SAM_Placebo_wk8         3.82650 0.909 414   4.210  0.0025
 # SAM_Amoxicillin_wk1 - SAMcontrol_NA_wk104     0.99401 1.594 409   0.624  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk12 -1.72735 0.941 356  -1.836  0.8582
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk4  -0.63870 0.968 351  -0.660  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Amoxicillin_wk8  -1.62441 0.996 341  -1.630  0.9370
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk0      -0.91685 1.005 413  -0.912  0.9997
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk1      -0.55801 0.999 413  -0.559  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk104    -0.86669 1.269 411  -0.683  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk12     -0.72433 0.993 413  -0.729  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk4      -0.07075 0.951 413  -0.074  1.0000
 # SAM_Amoxicillin_wk104 - SAM_Placebo_wk8      -0.46987 1.042 413  -0.451  1.0000
 # SAM_Amoxicillin_wk104 - SAMcontrol_NA_wk104  -3.30237 1.673 411  -1.974  0.7832
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk4    1.08865 0.817 353   1.332  0.9880
 # SAM_Amoxicillin_wk12 - SAM_Amoxicillin_wk8    0.10294 0.848 323   0.121  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk0        0.81049 0.857 414   0.946  0.9996
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk1        1.16934 0.849 414   1.377  0.9841
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk104      0.86066 1.155 413   0.745  1.0000
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk12       1.00302 0.842 414   1.191  0.9959
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk4        1.65659 0.793 414   2.090  0.7081
 # SAM_Amoxicillin_wk12 - SAM_Placebo_wk8        1.25747 0.900 414   1.397  0.9819
 # SAM_Amoxicillin_wk12 - SAMcontrol_NA_wk104   -1.57502 1.588 409  -0.992  0.9994
 # SAM_Amoxicillin_wk4 - SAM_Amoxicillin_wk8    -0.98571 0.879 325  -1.122  0.9977
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk0        -0.27815 0.888 414  -0.313  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk1         0.08069 0.881 414   0.092  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk104      -0.22799 1.179 412  -0.193  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk12       -0.08563 0.874 414  -0.098  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk4         0.56795 0.827 414   0.687  1.0000
 # SAM_Amoxicillin_wk4 - SAM_Placebo_wk8         0.16883 0.930 414   0.182  1.0000
 # SAM_Amoxicillin_wk4 - SAMcontrol_NA_wk104    -2.66367 1.606 410  -1.659  0.9287
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk0         0.70756 0.923 414   0.767  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk1         1.06640 0.916 414   1.164  0.9967
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk104       0.75772 1.205 412   0.629  1.0000
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk12        0.90008 0.909 414   0.990  0.9994
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk4         1.55365 0.864 414   1.799  0.8758
 # SAM_Amoxicillin_wk8 - SAM_Placebo_wk8         1.15453 0.963 414   1.199  0.9956
 # SAM_Amoxicillin_wk8 - SAMcontrol_NA_wk104    -1.67796 1.625 410  -1.033  0.9990
 # SAM_Placebo_wk0 - SAM_Placebo_wk1             0.35884 0.880 349   0.408  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk104           0.05017 1.166 337   0.043  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk12            0.19252 0.882 390   0.218  1.0000
 # SAM_Placebo_wk0 - SAM_Placebo_wk4             0.84610 0.834 380   1.014  0.9992
 # SAM_Placebo_wk0 - SAM_Placebo_wk8             0.44698 0.935 381   0.478  1.0000
 # SAM_Placebo_wk0 - SAMcontrol_NA_wk104        -2.38551 1.617 410  -1.476  0.9713
 # SAM_Placebo_wk1 - SAM_Placebo_wk104          -0.30867 1.171 373  -0.263  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk12           -0.16632 0.873 379  -0.191  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk4             0.48726 0.827 381   0.589  1.0000
 # SAM_Placebo_wk1 - SAM_Placebo_wk8             0.08814 0.925 367   0.095  1.0000
 # SAM_Placebo_wk1 - SAMcontrol_NA_wk104        -2.74436 1.613 410  -1.702  0.9145
 # SAM_Placebo_wk104 - SAM_Placebo_wk12          0.14235 1.175 398   0.121  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk4           0.79593 1.139 392   0.699  1.0000
 # SAM_Placebo_wk104 - SAM_Placebo_wk8           0.39681 1.215 393   0.327  1.0000
 # SAM_Placebo_wk104 - SAMcontrol_NA_wk104      -2.43568 1.793 413  -1.359  0.9858
 # SAM_Placebo_wk12 - SAM_Placebo_wk4            0.65358 0.818 371   0.799  0.9999
 # SAM_Placebo_wk12 - SAM_Placebo_wk8            0.25446 0.915 353   0.278  1.0000
 # SAM_Placebo_wk12 - SAMcontrol_NA_wk104       -2.57803 1.609 410  -1.602  0.9449
 # SAM_Placebo_wk4 - SAM_Placebo_wk8            -0.39912 0.871 349  -0.458  1.0000
 # SAM_Placebo_wk4 - SAMcontrol_NA_wk104        -3.23161 1.584 409  -2.041  0.7410
 # SAM_Placebo_wk8 - SAMcontrol_NA_wk104        -2.83249 1.640 410  -1.727  0.9053

#Now to evaluate only extended spectrum beta lactamases
# ESBL<-Abx_geneslong %>% filter(grepl('extended',Description))
# ESBL<-ESBL[-c(4:18)]
# ESBL<-reshape(ESBL, idvar="Sample", timevar = "Family", direction ="wide")
# ESBL[is.na(ESBL)] <- 0
# rownames(ESBL)=ESBL$Sample
# ESBL$Sample=NULL
# ESBL$Sum<-rowSums(ESBL)
# ESBL$Sample<-row.names(ESBL)
# write.csv(ESBL,"Shortbred_data/Data_tables/221019_ESBLs_RPKM.csv")
ESBL<-read.csv("Shortbred_data/Data_tables/221019_ESBLs_RPKM.csv")

metaESBL<-join(ESBL, metadata)
metaESBL$Treatment=factor(metaESBL$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
ggplot(metaESBL, aes(x=as.factor(Week_of_follow_up), y=Sum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Extended Spectrum Beta Lactamase Count")+xlab("Timepoint")+ylab("Beta lactamases") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))

#What about NON-BETA LACTAMASE resistance genes
# NONBetalactamases<-Abx_geneslong %>% filter(!grepl('Beta-lactamase',AMR_Classification))
# NONBetalactamases<-NONBetalactamases[-c(4:18)]
# NONBetalactamases<-reshape(NONBetalactamases, idvar="Sample", timevar = "Family", direction ="wide")
# NONBetalactamases[is.na(NONBetalactamases)] <- 0
# rownames(NONBetalactamases)=NONBetalactamases$Sample
# NONBetalactamases$Sample=NULL
# NONBetalactamases$Sum<-rowSums(NONBetalactamases)
# NONBetalactamases$Sample<-row.names(NONBetalactamases)
# write.csv(NONBetalactamases,"Shortbred_data/Data_tables/220620_NONbetalactamases_RPKM.csv")
NONBetalactamases<-read.delim("Shortbred_data/Data_tables/220620_NONbetalactamases_RPKM.csv", sep=",")
metaNONBeta<-join(metadata, NONBetalactamases)
metaNONBeta$Treatment=factor(metaNONBeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))

ggplot(metaNONBeta, aes(x=as.factor(Week_of_follow_up), y=Sum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("NONBeta Lactamase RPKM")+xlab("Timepoint")+ylab("NONBeta lactamases") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))+scale_y_continuous(trans = "log10")
#No real effect of removing beta lactamases on sum, what about count?

metaNONBeta<-join(metadata, NONBetalactamases)
ggplot(metaNONBeta, aes(x=as.factor(Week_of_follow_up), y=Count, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("NONBeta Lactamase genes per sample")+xlab("Timepoint")+ylab("NONBeta lactamases") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220714_NONBetalactamasecountovertime.pdf")

fitNONbetacount<- lmer(Count~Description+(1|idno), data=metaNONBeta)
summary(fitNONbetacount)
tukfitNONbetacount<-lsmeans(fitNONbetacount, pairwise~Description, adjust="Tukey")
tukfitNONbetacount
```

```{r}
#Maaslin on Metaphlan3 species
input_data_species <- read.delim("metaphlan3datatables/210621_wide_metaphlanStandard_species.txt", header = TRUE) #The abundance table file
input_data_genus<-read.delim("metaphlan3datatables/210621_wide_metaphlanStandard_genus.txt", header = TRUE)
input_data_family<-read.delim("metaphlan3datatables/210621_wide_metaphlanStandard_family.txt", header = TRUE)
input_data_phylum<-read.delim("metaphlan3datatables/210621_wide_metaphlanStandard_phyla.txt", header = TRUE)
input_metadata<-read.delim(sep=',',"210615_Nigermetadata.csv",header=TRUE,stringsAsFactors = F)
#Remove anything under 200K reads
input_metadata = input_metadata[which(input_metadata$Post.processed_readcount > 200000),]
input_metadata = input_metadata[which(!input_metadata$Treatment %in% c("Control")),]
#Remove 2 SAMcontrol with Z score >-3
input_metadata <- subset(input_metadata, idno!="7006" & idno!="7007")

input_metadata<-data.frame(input_metadata, row.names = 11)
input_data_species<-data.frame(input_data_species, row.names = 1)
input_data_genus<-data.frame(input_data_genus, row.names = 1)
input_data_family<-data.frame(input_data_family, row.names = 1)
input_data_phylum<-data.frame(input_data_phylum, row.names = 1)
input_ARGs<-read.delim("Shortbred_data/Data_tables/220818_Shortbredwide_wide_ShortbredID.csv", sep=",", header=TRUE)
input_ARGs<-data.frame(input_ARGs, row.names = 1)

#Species level
fit_data = Maaslin2(
    input_data_species,input_metadata, output = 'R_scripts/Maaslin/Description_Specieswk0plaref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk0',max_significance = 0.05)

fit_data = Maaslin2(
    input_data_species,input_metadata, output = 'R_scripts/Maaslin/Description_Specieswk104plaref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk104',max_significance = 0.05)

#Species level
fit_data = Maaslin2(
    input_data_species,input_metadata, output = 'R_scripts/Maaslin/Description_Specieshealthywk104ref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,Healthy_NA_wk104',max_significance = 0.05)

#Description at the genus level
fit_data = Maaslin2(
    input_data_genus,input_metadata, output = 'R_scripts/Maaslin/Description_Genuswk104plaref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.05)
fit_data = Maaslin2(
    input_data_genus,input_metadata, output = 'R_scripts/Maaslin/Description_Genushealthywk104ref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.05)

#Description at the family level
fit_data = Maaslin2(
    input_data_family,input_metadata, output = 'R_scripts/Maaslin/Description_Familywk104plaref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.05)

fit_data = Maaslin2(
    input_data_family,input_metadata, output = 'R_scripts/Maaslin/Description_Familywk104healthyref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.05)

#Description at the phylum level
fit_data = Maaslin2(
    input_data_phylum,input_metadata, output = 'R_scripts/Maaslin/Description_Phylumplawk104ref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.05)

fit_data = Maaslin2(
    input_data_phylum,input_metadata, output = 'R_scripts/Maaslin/Description_phylumhealthyref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,Healthy_NA_wk104',max_significance = 0.05)

#Interested in the Week 0 to 1 change
Week1meta<-subset(input_metadata, Week_of_follow_up==0|Week_of_follow_up==1)
fit_data = Maaslin2(
    input_data_species,Week1meta, output = 'R_scripts/Maaslin/Week0to1specieswk1plaref',
    fixed_effects = c('Description'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk1',max_significance = 0.05)

#Interested in the Week 0 to 1 change at genus level
fit_data = Maaslin2(
    input_data_genus,Week1meta, output = 'R_scripts/Maaslin/Week0to1genuswk1plaref',
    fixed_effects = c('Description','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk1',max_significance = 0.05)

fit_data = Maaslin2(
    input_data_genus,Week1meta, output = 'R_scripts/Maaslin/Week0to1genuswk0amoxref',
    fixed_effects = c('Description','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Amoxicillin_wk0',max_significance = 0.05)

#how about the differences between time 0 and time 104 for just placebo vs. amox?
Week104to0<-subset(input_metadata, Week_of_follow_up==0|Week_of_follow_up==104)
Week104to0<-subset(Week104to0, Treatment=="Amoxicillin"|Treatment=="Placebo")
Week104<-subset(input_metadata, Week_of_follow_up==104)
week104amoxpla<-subset(Week104, Treatment=="Amoxicillin"|Treatment=="Placebo")

fit_data = Maaslin2(
    input_data_species,Week104to0, output = 'R_scripts/Maaslin/Week0to104specieswk0plaref',
    fixed_effects = c('Description','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk0',max_significance = 0.05)

fit_data = Maaslin2(
    input_data_genus,Week104to0, output = 'R_scripts/Maaslin/Week0to104genuswk0plaref',
    fixed_effects = c('Description','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk0',max_significance = 0.05)

#Only Week 0, 1, and 104
Week0to1to104<-subset(input_metadata, Week_of_follow_up==0|Week_of_follow_up==1|Week_of_follow_up==104)
fit_data = Maaslin2(
    input_data_genus,Week0to1to104, output = 'R_scripts/Maaslin/Week0to1to104genuswk0ref',
    fixed_effects = c('Description','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk0',max_significance = 0.05)

fit_data = Maaslin2(
    input_data_species,Week104, output = 'R_scripts/Maaslin/Week104speciesplaref',
    fixed_effects = c('Description','visAge'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_genus,Week104, output = 'R_scripts/Maaslin/Week104genusplaref',
    fixed_effects = c('Description','visAge'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_family,Week104, output = 'R_scripts/Maaslin/Week104familyplaref',
    fixed_effects = c('Description','visAge'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_phylum,Week104, output = 'R_scripts/Maaslin/Week104phylumplaref',
    fixed_effects = c('Description','visAge'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.25)

fit_data = Maaslin2(
    input_data_species,Week104, output = 'R_scripts/Maaslin/Week104speciesphealthyref',
    fixed_effects = c('Description','visAge'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_species,Week104, output = 'R_scripts/Maaslin/Week104speciesSAMref',
    fixed_effects = c('Description','visAge'), reference = 'Description,SAMcontrol_NA_wk104', max_significance = 0.25)

fit_data = Maaslin2(
    input_data_genus,Week104, output = 'R_scripts/Maaslin/Week104genushealthyref',
    fixed_effects = c('Description','visAge'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_family,Week104, output = 'R_scripts/Maaslin/Week104familyhealthyref',
    fixed_effects = c('Description','visAge'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.25)
fit_data = Maaslin2(
    input_data_phylum,Week104, output = 'R_scripts/Maaslin/Week104phylumhealthyref',
    fixed_effects = c('Description','visAge'), reference = 'Description,Healthy_NA_wk104', max_significance = 0.25)

#Just week 104 what does age do with healthy samples
healthy<-subset(Week104, Treatment=="Healthy")
fit_data = Maaslin2(
    input_data_genus,healthy, output = 'R_scripts/Maaslin/Week104healthygenus',
    fixed_effects = c('visAge','visMUAC','visWHZ'), max_significance = 0.05)
fit_data = Maaslin2(
    input_data_family,healthy, output = 'R_scripts/Maaslin/Week104healhyfamily',
    fixed_effects = c('visAge','visMUAC','visWHZ'), max_significance = 0.05)
fit_data = Maaslin2(
    input_data_phylum,healthy, output = 'R_scripts/Maaslin/Week104healhyphylum',
    fixed_effects = c('visAge','visMUAC','visWHZ'), max_significance = 0.05)
fit_data = Maaslin2(
    input_data_species,healthy, output = 'R_scripts/Maaslin/Week104healhyspecies',
    fixed_effects = c('visAge','visMUAC','visWHZ'), max_significance = 0.05)

#how about the differences between time 1 and time 104 for just placebo vs. amox?
Week104to1<-subset(input_metadata, Week_of_follow_up==1|Week_of_follow_up==104)
Week104to1<-subset(Week104to1, Treatment=="Amoxicillin"|Treatment=="Placebo")
fit_data = Maaslin2(
    input_data_genus,Week104to1, output = 'R_scripts/Maaslin/Week1to104genus',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'),reference = 'Description,SAM_Placebo_wk1',)

library(scales)

#What about the first 12 weeks 
first12week<-subset(input_metadata, Week_of_follow_up<14)
#phylum level
fit_data = Maaslin2(
    input_data_phylum,first12week, output = 'R_scripts/Maaslin/First12weeksPhylum',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk0', max_significance = 0.25)
#Family level
fit_data = Maaslin2(
    input_data_family,first12week, output = 'R_scripts/Maaslin/First12weeksFamily',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk0', max_significance = 0.25)
#Genus level
fit_data = Maaslin2(
    input_data_genus,first12week, output = 'R_scripts/Maaslin/First12weeksGenus',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk0', max_significance = 0.25)
#Species level
fit_data = Maaslin2(
    input_data_species,first12week, output = 'R_scripts/Maaslin/First12weeksSpecies',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk0', max_significance = 0.25)

#Just the amoxicillin and placebo groups over time
amoxipla<-subset(input_metadata, Treatment=="Amoxicillin"|Treatment=="Placebo")
fit_data = Maaslin2(
    input_data_species,amoxipla, output = 'R_scripts/Maaslin/justamoxplaspecies',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = 'Description,SAM_Placebo_wk104', max_significance = 0.05)

fit_data = Maaslin2(
    input_data_genus,amoxipla, output = 'R_scripts/Maaslin/justamoxplagenusplacebo104ref',
    fixed_effects = c('Description','visMUAC','visWHZ','visdelMUAC','visdelWHZ','visAge'), random_effects = c('idno'), reference = "Description,SAM_Placebo_wk104", max_significance = 0.05)

#Just to make sure no difference in week 0
Week0<-subset(input_metadata, Week_of_follow_up==0)
fit_data = Maaslin2(
    input_data_genus,Week0, output = 'R_scripts/Maaslin/Week0only',
    fixed_effects = c('Description','visAge'), max_significance = 0.05)
```

```{r}
#Plot significant results as coefficient plot. 
wk0to1genusmaaslinsigwk0amoxref <- read.csv('R_scripts/Maaslin/Week0to1genuswk0amoxref/wk0to1genussignificant_resultswk0amoxref.tsv', sep='')
wk0to1genusmaaslinsigwk1plaref<-read.csv('R_scripts/Maaslin/Week0to1genuswk1plaref/wk0to1genussignificant_resultswk1plaref.tsv', sep='')
wk0to1genussigcombined<-rbind(wk0to1genusmaaslinsigwk0amoxref,wk0to1genusmaaslinsigwk1plaref)
#Description only
wk0to1genussigcombinedDescription<-subset(wk0to1genussigcombined, metadata=="Description")
#Sort by descending coefficient (doesn't actually relevel factors though)
wk0to1genussigcombinedDescription <- wk0to1genussigcombinedDescription %>% arrange(desc(coef))

#COEFFICIENT PLOT
wk0to1genussigcombinedDescription$CI95upper <- wk0to1genussigcombinedDescription$coef + 1.96*wk0to1genussigcombinedDescription$stderr
wk0to1genussigcombinedDescription$CI95lower <- wk0to1genussigcombinedDescription$coef - 1.96*wk0to1genussigcombinedDescription$stderr
wk0to1genussigcombinedDescription$Interaction<-paste(wk0to1genussigcombinedDescription$feature,wk0to1genussigcombinedDescription$ref)

wk0to1genussigcombinedDescription$feature=factor(wk0to1genussigcombinedDescription$feature, levels=wk0to1genussigcombinedDescription$feature[order(wk0to1genussigcombinedDescription$coef)])

wk0to1genussigcombinedDescription$feature<-factor(wk0to1genussigcombinedDescription$feature, levels=c("g__Streptococcus","g__Lactobacillus","g__Bifidobacterium","g__Dorea","g__Holdemanella","g__Klebsiella","g__Escherichia"))

ggplot(wk0to1genussigcombinedDescription, aes(x=coef, y=feature, color=as.factor(coef<0)))+
  geom_point()+facet_grid(cols = vars(ref))+
  geom_errorbarh(aes(xmax = CI95upper, xmin = CI95lower, height =0.1))+
  geom_vline(xintercept = 0, linetype='dotted', color='grey', size=1.5)+
  #scale_color_brewer(palette ='Dark2')+
  theme_classic()+
  theme(axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=15),
        axis.title.x = element_text(size=15),
        axis.title.y = element_blank(),
        legend.position = "none",
        plot.title = element_text(size=15),
        panel.border = element_rect(fill=NA, colour="black", size=3))+
  xlab('coefficient')+scale_color_manual(values = c("#E31A1C","#1F78B4"))
ggsave('220805_week0to1genussignificantall.pdf')
# write.csv(diffspecies.a.df2, '220612_Mphln3_NEGBIN/220612_diffspecies_coeff.csv')

#Now plotting overall differences referrent to week 104 placebo
significantresultsweek104plaref<-read.csv('R_scripts/Maaslin/Description_Specieswk104plaref/significant_resultswk104plaref.tsv', sep='')

#Just week 104 amox
week104amoxsigspecies<-subset(significantresultsweek104plaref, value=="SAM_Amoxicillin_wk104")
week104amoxsigspecies$CI95upper <- week104amoxsigspecies$coef + 1.96*week104amoxsigspecies$stderr
week104amoxsigspecies$CI95lower <- week104amoxsigspecies$coef - 1.96*week104amoxsigspecies$stderr
week104amoxsigspecies$feature=factor(week104amoxsigspecies$feature, levels=week104amoxsigspecies$feature[order(week104amoxsigspecies$coef)])

ggplot(week104amoxsigspecies, aes(x=coef, y=feature, color=as.factor(coef<0)))+
  geom_point()+
  geom_errorbarh(aes(xmax = CI95upper, xmin = CI95lower, height =0.1))+
  geom_vline(xintercept = 0, linetype='dotted', color='grey', size=1.5)+
  #scale_color_brewer(palette ='Dark2')+
  theme_classic()+
  theme(axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=15),
        axis.title.x = element_text(size=15),
        axis.title.y = element_blank(),
        legend.position = "none",
        plot.title = element_text(size=15),
        panel.border = element_rect(fill=NA, colour="black", size=3))+
  xlab('coefficient')+scale_color_manual(values = c("#E31A1C","#1F78B4"))
ggsave('220818_week104amoxtoplacebosigspecies.pdf')
# write.csv(diffspecies.a.df2, '220612_Mphln3_NEGBIN/220612_diffspecies_coeff.csv')

#Now plotting overall differences referrent to week 104 nonSAM
significantresultsweek104healthyref<-read.csv('R_scripts/Maaslin/Description_Specieshealthywk104ref/significant_resultswk104healthyref.tsv', sep='')

#Just week 104 placebo
week104plasigspecies<-subset(significantresultsweek104healthyref, value=="SAM_Placebo_wk104")
week104plasigspecies$CI95upper <- week104plasigspecies$coef + 1.96*week104plasigspecies$stderr
week104plasigspecies$CI95lower <- week104plasigspecies$coef - 1.96*week104plasigspecies$stderr
week104plasigspecies$feature=factor(week104plasigspecies$feature, levels=week104plasigspecies$feature[order(week104plasigspecies$coef)])

ggplot(week104plasigspecies, aes(x=coef, y=feature, color=as.factor(coef<0)))+
  geom_point()+
  geom_errorbarh(aes(xmax = CI95upper, xmin = CI95lower, height =0.1))+
  geom_vline(xintercept = 0, linetype='dotted', color='grey', size=1.5)+
  #scale_color_brewer(palette ='Dark2')+
  theme_classic()+
  theme(axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=15),
        axis.title.x = element_text(size=15),
        axis.title.y = element_blank(),
        legend.position = "none",
        plot.title = element_text(size=15),
        panel.border = element_rect(fill=NA, colour="black", size=3))+
  xlab('coefficient')+scale_color_manual(values = c("#1F78B4","#00FF00"))
ggsave('221016_week104placebotohealthysigspecies.pdf')

```

```{r}
#PLot significant associations from Maaslin
widespeciesmeta<-join(metadata, Specieswide)
widegenusmeta<-join(metadata, Genuswide)
widefamilymeta<-join(metadata, Familywide)
widephylummeta<-join(metadata, Phylumwide)
write.csv(widespeciesmeta, "metaphlan3datatables/220619_widespeciesmetadata.csv")

#Firmicutes/bacteroidetes ratio
widephylummeta$FtoBratio<-widephylummeta$p__Bacteroidetes/widephylummeta$p__Firmicutes


widespeciesmeta$Treatment=factor(widespeciesmeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
widegenusmeta$Treatment=factor(widegenusmeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
widephylummeta$Treatment=factor(widephylummeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))
widefamilymeta$Treatment=factor(widefamilymeta$Treatment, levels = c("Amoxicillin", "Placebo", "Healthy", "SAMcontrol"))


#week 0 and 1  only at genus level and only week 1 and 104
Genuswk0to1<-subset(widegenusmeta, Week_of_follow_up=="0"|Week_of_follow_up=="1")
Genuswk1to104<-subset(widegenusmeta, Week_of_follow_up=="1"|Week_of_follow_up=="104")
Genuswk1to104<-subset(Genuswk1to104,Treatment=="Amoxicillin"|Treatment=="Placebo")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Fusicatenibacter, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Richness")+xlab("Timepoint")+ylab("Fusicatenibacter abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220602_Fusicatenibacter.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Escherichia, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Escherichia")+xlab("Timepoint")+ylab("Escherichia abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220602_Escherichia.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Escherichia, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Escherichia")+xlab("Timepoint")+ylab("Escherichia abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220615_Escherichiaweek0to1.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Lactobacillus, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Lactobacillus")+xlab("Timepoint")+ylab("Lactobacillus abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220615_Lctobacillusweek0to1.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Bifidobacterium, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Bifidobacterium")+xlab("Timepoint")+ylab("Bifidobacterium abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220615_Bifidobacteriumweek0to1.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Klebsiella, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Klebsiella")+xlab("Timepoint")+ylab("Klebsiella abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220620_Klebsiellaweek0to1.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Klebsiella, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Klebsiella")+xlab("Timepoint")+ylab("Klebsiella abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220620_Klebsiellaweek0to1.pdf")

ggplot(Genuswk0to1, aes(x=as.factor(Week_of_follow_up), y=g__Holdemanella, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Holdemanella")+xlab("Timepoint")+ylab("Holdemanella abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220620_Holdemanellaweek0to1.pdf")


ggplot(Genuswk1to104, aes(x=as.factor(Week_of_follow_up), y=g__Holdemanella, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Holdemanella")+xlab("Timepoint")+ylab("Holdemanella abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4"))
ggsave(file="2021Figs/220620_Holdemanellaweek1to104.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Actinomyces, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Actinomyces")+xlab("Timepoint")+ylab("Actinomyces abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Actinomycesovertime.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Streptococcus, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Streptococcus")+xlab("Timepoint")+ylab("Streptococcus abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Streptococcusovertime.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Bifidobacterium, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Bifidobacterium")+xlab("Timepoint")+ylab("Bifidobacterium abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Bifidobacteriumovertime.pdf")

#Abundance of Bifidobacterium genus at week 104 ANOVA
pairwise.wilcox.test(week104genus$g__Bifidobacterium, week104genus$Treatment, p.adjust.method = "BH")
#            Amoxicillin Healthy Placebo
# Healthy    0.0084      -       -      
# Placebo    0.2119      0.3590  -      
# SAMcontrol 0.0122      0.2949  0.2119 
# 
# P value adjustment method: BH

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Lactobacillus, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Lactobacillus")+xlab("Timepoint")+ylab("Lactobacillus abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Lactobacillusovertime.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Prevotella, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Prevotella")+xlab("Timepoint")+ylab("Prevotella abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Prevotellaovertime.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Fusicatenibacter, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Fusicatenibacter")+xlab("Timepoint")+ylab("Fusicatenibacter abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Fusicatenibacterovertime.pdf")

ggplot(widegenusmeta, aes(x=as.factor(Week_of_follow_up), y=g__Faecalibacterium, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Faecalibacterium")+xlab("Timepoint")+ylab("Faecalibacterium abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Faecalibacteriumovertime.pdf")

ggplot(widephylummeta, aes(x=as.factor(Week_of_follow_up), y=p__Firmicutes, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Firmicutes")+xlab("Timepoint")+ylab("Firmicutes abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Firmicutesovertime.pdf")

ggplot(widephylummeta, aes(x=as.factor(Week_of_follow_up), y=p__Actinobacteria, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Actinobacteria")+xlab("Timepoint")+ylab("Actinobacteria abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Actinobacteriaovertime.pdf")

ggplot(widephylummeta, aes(x=as.factor(Week_of_follow_up), y=p__Bacteroidetes, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Bacteroidetes")+xlab("Timepoint")+ylab("Bacteroidetes abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220620_Bacteroidetesovertime.pdf")

#Just week 104
week104species<-subset(widespeciesmeta, Week_of_follow_up==104)
week104genus<-subset(widegenusmeta, Week_of_follow_up==104)
week104family<-subset(widefamilymeta, Week_of_follow_up==104)
week104phylum<-subset(widephylummeta, Week_of_follow_up==104)
ggplot(week104species, aes(x=as.factor(Treatment), y=s__Prevotella_sp_885, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Prevotella_sp_885")+xlab("Treatment")+ylab("Prevotella_sp_885 abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104Provotella_sp_885.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Firmicutes_bacterium_CAG_791, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Firmicutes_bacterium_CAG_791")+xlab("Treatment")+ylab("s__Firmicutes_bacterium_CAG_791 abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Firmicutes_bacterium_CAG_791.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Brachyspira_sp_CAG_700, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Brachyspira_sp_CAG_700")+xlab("Treatment")+ylab("Brachyspira_sp_CAG_700 abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104Brachyspira_sp_CAG_700.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Bifidobacterium_bifidum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Bifidobacterium_bifidum")+xlab("Treatment")+ylab("s__Bifidobacterium_bifidum abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Bifidobacterium_bifidum.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Eubacterium_sp_CAG_251, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Eubacterium_sp_CAG_251")+xlab("Treatment")+ylab("s__Eubacterium_sp_CAG_251 abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Eubacterium_sp_CAG_251.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Clostridium_perfringens, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Clostridium_perfringens")+xlab("Treatment")+ylab("s__Clostridium_perfringens abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Clostridium_perfringens.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Coprococcus_eutactus, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Coprococcus_eutactus")+xlab("Treatment")+ylab("s__Coprococcus_eutactus abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Coprococcus_eutactus.pdf")

ggplot(week104species, aes(x=as.factor(Treatment), y=s__Weissella_confusa, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("s__Weissella_confusa")+xlab("Treatment")+ylab("s__Weissella_confusa abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220717_Week104s__Weissella_confusa.pdf")


ggplot(widephylummeta, aes(x=as.factor(Week_of_follow_up), y=FtoBratio, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Firmicutes to Bacteroidetes ratio")+xlab("Timepoint")+scale_y_log10()+ylab("Firmicutes to Bacteroidetes ratio") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220721_FirmicutestoBacteroidetesratio.pdf")

ggplot(week104family, aes(x=as.factor(Week_of_follow_up), y=f__Actinomycetaceae, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Actinomycetaceae")+xlab("Timepoint")+ylab("Actinomycetaceae") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220724_Actinomycetaceae.pdf")

ggplot(widefamilymeta, aes(x=as.factor(Week_of_follow_up), y=f__Campylobacteraceae, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Campylobacteraceae")+xlab("Timepoint")+ylab("Campylobacteraceae abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220807_Campylobacteraceae.pdf")

#Plotting significant species over time from Maaslin relative to week 104 placebo
ggplot(widespeciesmeta, aes(x=as.factor(Week_of_follow_up), y=s__Bifidobacterium_longum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Bifidobacterium longum")+xlab("Timepoint")+ylab("Bifidobacterium longum abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220818_Bifidobacteriumlongum.pdf")

ggplot(widespeciesmeta, aes(x=as.factor(Week_of_follow_up), y=s__Bifidobacterium_bifidum, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Bifidobacterium bifidum")+xlab("Timepoint")+ylab("Bifidobacterium bifidum abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220818_Bifidobacteriumbifidum.pdf")

ggplot(widespeciesmeta, aes(x=as.factor(Week_of_follow_up), y=s__Weissella_confusa, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Weissella confusa")+xlab("Timepoint")+ylab("Weisella confusa abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220818_Weissellaconfusa.pdf")

ggplot(widespeciesmeta, aes(x=as.factor(Week_of_follow_up), y=s__Prevotella_sp_885, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Prevotella sp. 885")+xlab("Timepoint")+ylab("Prevotella sp. 885 abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220818_Prevotellasp.885.pdf")

ggplot(widespeciesmeta, aes(x=as.factor(Week_of_follow_up), y=s__Prevotella_stercorea, color=Treatment)) + geom_boxplot(outlier.shape = NA) +geom_point(position = position_jitterdodge(jitter.width = 0.1)) +ggtitle("Prevotella stercorea")+xlab("Timepoint")+ylab("Prevotella stercorea abundance") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+theme(axis.text.x = element_text(size=16), axis.title.x=element_text(size=20), axis.title.y = element_text(size=20), axis.text.y = element_text(size=16), axis.title = element_text(size=24))+scale_color_manual(values = c("Amoxicillin" = "#E31A1C", "Placebo" = "#1F78B4","SAMcontrol" = "#FF7F00", "Healthy" = "#33A02C"))
ggsave(file="2021Figs/220818_Prevotella_stercorea.pdf")
```

```{r}
#RandomForest model of microbiota development of healthy kids
Healthy<-subset(metadata, Treatment=='Healthy')
Healthyspecies<-join(Healthy, Specieswide)
Healthygenus<-join(Healthy, Genuswide)

randomForest(formula=Healthyspecies$SAGE~ ., data=Healthyspecies[,213:659], ntree=10000, importance=TRUE)
# Call:
#  randomForest(formula = Healthyspecies$SAGE ~ ., data = Healthyspecies[,      213:659], ntree = 10000, importance = TRUE) 
#                Type of random forest: regression
#                      Number of trees: 10000
# No. of variables tried at each split: 149
# 
#           Mean of squared residuals: 107.0626
#                     % Var explained: 14.25
Healthyspecies$prediction<-predict(randomForest(formula=Healthyspecies$SAGE~ ., data=Healthyspecies[,213:659], ntree=10000, importance=TRUE))
randomForest(formula=Healthygenus$SAGE~ ., data=Healthygenus[,213:356], ntree=10000, importance=TRUE)
Healthyimportance<- importance(randomForest(formula=Healthyspecies$visAge~ ., data=Healthyspecies[,213:659], ntree=10000, importance=TRUE))
Healthyimportance<-as.data.frame(Healthyimportance)
Healthyimportance$Feature<-row.names(Healthyimportance)
#only positive features above 5%
Healthyimportancepositive<-subset(Healthyimportance, `%IncMSE`>=5)
write.csv(Healthyimportancepositive, "2021Figs/221016_Healthyimportance.csv")
Healthyimportancepositive$Feature=factor(Healthyimportancepositive$Feature, levels=Healthyimportancepositive$Feature[order(Healthyimportancepositive$`%IncMSE`)])
ggplot(Healthyimportancepositive)+ geom_point(aes(x=`%IncMSE`, y=Feature))
ggsave(file="2021Figs/221016_HEALTHYFeaturevsMSE.pdf")

ggplot(Healthyspecies, aes(x=SAGE, y=prediction))+geom_point()+geom_smooth(method='lm',formula=y~x)
ggsave(file="2021Figs/220604_HealthyAgevspredictedage.pdf")
summary(lm(prediction~SAGE, data=Healthyspecies))
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -9.5889 -3.2535 -0.6374  1.9911 10.5471 
# 
# Coefficients:
#             Estimate Std. Error t value Pr(>|t|)    
# (Intercept) 13.08343    1.34559   9.723 1.31e-11 ***
# SAGE         0.16683    0.06701   2.490   0.0175 *  
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 4.616 on 36 degrees of freedom
# Multiple R-squared:  0.1469,	Adjusted R-squared:  0.1232 
# F-statistic: 6.198 on 1 and 36 DF,  p-value: 0.01755


#Pretty poor variance explained with the above models. Maybe try to focus on the under 30MOL samples
Healthyspeciesunder30<-subset(Healthyspecies, SAGE<=30)
Healthygenusunder30<-subset(Healthygenus, SAGE<=30)
randomForest(formula=Healthyspeciesunder30$SAGE~ ., data=Healthyspeciesunder30[,213:659], ntree=10000, importance=TRUE)
randomForest(formula=Healthygenusunder30$SAGE~ ., data=Healthygenusunder30[,213:356], ntree=10000, importance=TRUE)

#Now will try to build it on the placebo group over time
Placebo<-subset(metadata, Treatment=='Placebo')
Placebospecies<-join(Placebo, Specieswide)
randomForest(formula=Placebospecies$visAge~ ., data=Placebospecies[,213:659], ntree=10000, importance=TRUE)
#  Mean of squared residuals: 56.59679
# % Var explained: 47.2
Placebospecies$prediction<-predict(randomForest(formula=Placebospecies$visAge~ ., data=Placebospecies[,213:659], ntree=10000, importance=TRUE))
Placeboimportance<- importance(randomForest(formula=Placebospecies$visAge~ ., data=Placebospecies[,213:659], ntree=10000, importance=TRUE))
Placeboimportance<-as.data.frame(Placeboimportance)
Placeboimportance$Feature<-row.names(Placeboimportance)
#only positive features above 5%
Placeboimportancepositive<-subset(Placeboimportance, `%IncMSE`>=5)
write.csv(Placeboimportance, "2021Figs/220604_PlaceboRFimportance.csv")

Placeboimportancepositive$Feature=factor(Placeboimportancepositive$Feature, levels=Placeboimportancepositive$Feature[order(Placeboimportancepositive$`%IncMSE`)])
ggplot(Placeboimportancepositive)+ geom_point(aes(x=`%IncMSE`, y=Feature))
ggsave(file="2021Figs/220613_FeaturevsMSE.pdf")


ggplot(Placebospecies, aes(x=visAge, y=prediction))+geom_point()+geom_smooth(method='lm',formula=y~x)
ggsave(file="2021Figs/220604_PlaceboAgevspredictedage.pdf")
summary(lm(prediction~visAge, data=Placebospecies))
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -18.0476  -3.0376  -0.7315   2.6691  14.4065 
# 
# Coefficients:
#             Estimate Std. Error t value Pr(>|t|)    
# (Intercept)  9.12457    0.70784   12.89   <2e-16 ***
# visAge       0.44868    0.03506   12.80   <2e-16 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 4.897 on 180 degrees of freedom
# Multiple R-squared:  0.4764,	Adjusted R-squared:  0.4735 
# F-statistic: 163.7 on 1 and 180 DF,  p-value: < 2.2e-16

#Now to try to run rfcv to get cross-validation on placebo species over time
rfcv10<-rfcv(trainx = Placebospecies[,213:659], trainy = Placebospecies$visAge, cv.fold = 10, step = 0.5)


rfcv10data<-cbind(rfcv5$n.var,as.data.frame(rfcv5$error.cv))
colnames(rfcv10data)<-c("nvar","error")
ggplot(rfcv10data, aes(x=nvar, y=error))+geom_point()
ggsave(file="2021Figs/220614_PlaceboRFerrorvsnvar10fold.pdf")


#What about combining species, ARGs, and pathways?
HealthyALL<-join(Healthy, ABX_wide)
HealthyALL<-join(HealthyALL, Specieswide)
HealthyALL<-join(HealthyALL, Humann3_pathwayabundance)

randomForest(formula=HealthyALL$SAGE~ ., data=HealthyALL[,213:1678], ntree=10000, importance=TRUE)
#Mean of squared residuals: 113.4006
#% Var explained: 9.17

#Try to subset placebo group to only one sample per ID
write.csv(Placebospecies, "Placebospecies.csv")
Placebosubsample<- Placebospecies %>% group_by(idno) %>% sample_n(1)
randomForest(formula=Placebosubsample$visAge~ ., data=Placebosubsample[,213:659], ntree=10000, importance=TRUE)
#  Mean of squared residuals: 37.97578
#                    % Var explained: 53.04
Placebosubsample$prediction<-predict(randomForest(formula=Placebosubsample$visAge~ ., data=Placebosubsample[,213:659], ntree=10000, importance=TRUE))
Placebosubsampleimportance<-importance(randomForest(formula=Placebosubsample$visAge~ ., data=Placebosubsample[,213:659], ntree=10000, importance=TRUE))
Placebosubsampleimportance<-as.data.frame(Placebosubsampleimportance)
Placebosubsampleimportance$Feature<-row.names(Placebosubsampleimportance)

#only positive features above 5%
Placebosubsampleimportancepositive<-subset(Placebosubsampleimportance, `%IncMSE`>=5)
write.csv(Placebosubsampleimportance, "2021Figs/220630_PlacebosubsampleRFimportance.csv")

Placebosubsampleimportancepositive$Feature=factor(Placebosubsampleimportancepositive$Feature, levels=Placebosubsampleimportancepositive$Feature[order(Placebosubsampleimportancepositive$`%IncMSE`)])
ggplot(Placebosubsampleimportancepositive)+ geom_point(aes(x=`%IncMSE`, y=Feature))
ggsave(file="2021Figs/220630_PlacebosubsampleimportancepositiveFeaturevsMSE.pdf")

ggplot(Placebosubsample, aes(x=visAge, y=prediction))+geom_point()+geom_smooth(method='lm',formula=y~x)
ggsave(file="2021Figs/220630_PlacebosubsampleAgevspredictedage.pdf")
summary(lm(prediction~visAge, data=Placebosubsample))

rfcv10subsample<-rfcv(trainx = Placebosubsample[,213:659], trainy = Placebosubsample$visAge, cv.fold = 10, step = 0.5)

rfcv10subsampleata<-cbind(rfcv10subsample$n.var,as.data.frame(rfcv10subsample$error.cv))
colnames(rfcv10subsampleata)<-c("nvar","error")
ggplot(rfcv10subsampleata, aes(x=nvar, y=error))+geom_point()
ggsave(file="2021Figs/220630_PlacebosubsampleRFerrorvsnvar10fold.pdf")

#The above was a random subsampling, but given the age distributions I want to only use the oldest kids. Subsampled the Placebo group to contain no duplicates and always chose the older sample
PlaceboOLD<-read.delim(sep=',',"Placebospecies_subsampleOLD.csv",header=TRUE,stringsAsFactors = F)
randomForest(formula=PlaceboOLD$visAge~ ., data=PlaceboOLD[,213:659], ntree=10000, importance=TRUE)
#40.8% variance explained
PlaceboOLD$prediction<-predict(randomForest(formula=PlaceboOLD$visAge~ ., data=PlaceboOLD[,213:659], ntree=10000, importance=TRUE))
PlaceboOLDimportance<-importance(randomForest(formula=PlaceboOLD$visAge~ ., data=PlaceboOLD[,213:659], ntree=10000, importance=TRUE))
PlaceboOLDimportance<-as.data.frame(PlaceboOLDimportance)
PlaceboOLDimportance$Feature<-row.names(PlaceboOLDimportance)

#only positive features above 5%
PlaceboOLDimportancepositive<-subset(PlaceboOLDimportance, `%IncMSE`>=5)
write.csv(PlaceboOLDimportance, "2021Figs/220630_PlaceboOLDRFimportance.csv")

PlaceboOLDimportancepositive$Feature=factor(PlaceboOLDimportancepositive$Feature, levels=PlaceboOLDimportancepositive$Feature[order(PlaceboOLDimportancepositive$`%IncMSE`)])
ggplot(PlaceboOLDimportancepositive)+ geom_point(aes(x=`%IncMSE`, y=Feature))
ggsave(file="2021Figs/220630_PlaceboOLDimportancepositiveFeaturevsMSE.pdf")

ggplot(PlaceboOLD, aes(x=visAge, y=prediction))+geom_point()+geom_smooth(method='lm',formula=y~x)
ggsave(file="2021Figs/220630_PlaceboOLDAgevspredictedage.pdf")
summary(lm(prediction~visAge, data=PlaceboOLD))

rfcv10OLD<-rfcv(trainx = PlaceboOLD[,213:659], trainy = PlaceboOLD$visAge, cv.fold = 10, step = 0.5)

rfcv10OLData<-cbind(rfcv10OLD$n.var,as.data.frame(rfcv10OLD$error.cv))
colnames(rfcv10OLData)<-c("nvar","error")
ggplot(rfcv10OLData, aes(x=nvar, y=error))+geom_point()
ggsave(file="2021Figs/220630_PlaceboOLDRFerrorvsnvar10fold.pdf")

```
